<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="borangao" />

<meta name="date" content="2023-10-07" />

<title>Data Preparation</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MESuSiE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="installation.html">Introduction</a>
</li>
<li>
  <a href="GWAS_QC.html">Data Preparation</a>
</li>
<li>
  <a href="Run_MESuSiE.html">Run MESuSiE</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis reproduce
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Simulation.html">Simulation</a>
    </li>
    <li>
      <a href="Lipid_711.html">711 Regions of Lipid</a>
    </li>
    <li>
      <a href="Lipid_810.html">810 Regions of Lipid</a>
    </li>
    <li>
      <a href="MCHC_SCZ.html">MCHC and SCZ</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/borangao/meSuSie_Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Data Preparation</h1>
<h4 class="author">borangao</h4>
<h4 class="date">2023-10-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-01-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>meSuSie_Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220530code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220530code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220530)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/net/fantasia/home/borang/Susie_Mult/meSuSie_Analysis/
</td>
<td style="text-align:left;">
.
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/fantasia/home/borang/Susie_Mult/meSuSie_Analysis/data/MESuSiE_Example.RData
</td>
<td style="text-align:left;">
data/MESuSiE_Example.RData
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomborangaomeSuSieAnalysistreea7405510738f7f4711fb915d43f4bb1998519691targetblanka740551a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/borangao/meSuSie_Analysis/tree/a7405510738f7f4711fb915d43f4bb1998519691" target="_blank">a740551</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomborangaomeSuSieAnalysistreea7405510738f7f4711fb915d43f4bb1998519691targetblanka740551a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/borangao/meSuSie_Analysis/tree/a7405510738f7f4711fb915d43f4bb1998519691" target="_blank">a740551</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  data/GLGC_chr_22.txt
    Untracked:  data/MESuSiE_Example.RData
    Untracked:  data/UKBB_chr_22.txt

Unstaged changes:
    Deleted:    analysis/illustration.Rmd
    Deleted:    analysis/toy_example.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/GWAS_QC.Rmd</code>) and HTML
(<code>docs/GWAS_QC.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/a7405510738f7f4711fb915d43f4bb1998519691/analysis/GWAS_QC.Rmd" target="_blank">a740551</a>
</td>
<td>
borangao
</td>
<td>
2024-01-15
</td>
<td>
Update GWAS QC
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/borangao/meSuSie_Analysis/428bdf044282b852602a3441b6bd45c13d6fdc1f/docs/GWAS_QC.html" target="_blank">428bdf0</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/ba8db567b85c23b12dd18456c89353b3fcfc63a2/analysis/GWAS_QC.Rmd" target="_blank">ba8db56</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Update my analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/borangao/meSuSie_Analysis/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/GWAS_QC.html" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/62ce4b3f045d8a95de6224cb4bca6e2890f28abc/analysis/GWAS_QC.Rmd" target="_blank">62ce4b3</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Update my analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-preparation-for-multiple-ancestry-fine-mapping"
class="section level1">
<h1>Data Preparation for Multiple-Ancestry Fine-Mapping</h1>
<blockquote>
<p><strong>Note:</strong> Following this guideline will help maintain
the integrity and accuracy of the fine-mapping process across multiple
ancestries.</p>
</blockquote>
<div id="step-by-step-guideline" class="section level2">
<h2>Step-by-Step Guideline</h2>
<p>For illustrative purposes, we utilized the GWAS summary statistics of
HDL on chromosome 22 from the UK Biobank (UKBB; <a
href="https://www.nealelab.is/uk-biobank"
class="uri">https://www.nealelab.is/uk-biobank</a>) and the Global
Lipids Genetics Consortium (GLGC; <a
href="https://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/"
class="uri">https://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/</a>).
We offer a suite of functions for data preparation, which include
<code>GWAS_QC</code>,<code>find_common_snps</code>,<code>allele_flip</code>.
Additionally, we have adapted the <code>kriging_rss</code> function from
SuSiE for enhanced functionality.</p>
<div id="gwas-preparations" class="section level3">
<h3>1. GWAS Preparations:</h3>
<ul>
<li><strong>Uniform QC:</strong> Apply a uniform QC process for each
individual ancestry.</li>
</ul>
<p>We advise removing strand-ambiguous SNPs, multi-allelic SNPs, SNPs
within MHC regions, and SNPs with a Minor Allele Frequency (MAF) less
than 0.001. The <code>GWAS_QC</code> function requires summary data
columns: CHR, POS, CHR_POS, REF, ALT, MAF, BETA, and SE. Additionally,
users have the flexibility to define a specific MAF threshold when using
the function.</p>
<pre class="r"><code>library(MESuSiE)
library(data.table)
library(dplyr)
library(snpStats)

setwd(&quot;/net/fantasia/home/borang/Susie_Mult/meSuSie_Analysis/&quot;)
UKBB &lt;- fread(&quot;data/UKBB_chr_22.txt&quot;)
GLGC &lt;- fread(&quot;data/GLGC_chr_22.txt&quot;)
UKBB &lt;- GWAS_QC(UKBB, 0.001)
GLGC &lt;- GWAS_QC(GLGC, 0.001)</code></pre>
<ul>
<li><strong>Subset of SNPs:</strong> Start with a consistent subset of
SNPs from various ancestries. We used UKBB European and African ancestry
genotype data as reference panel in the analysis. For users who do not
have access to UKBB data, you can use genotype data of 1,000 Genome
Project.</li>
</ul>
<p>Here we use <code>find_common_snps</code> to subset a common set of
SNPs across ancestries. We matched up by CHR_POS and also check the
reference and alternative allele are matched up together.</p>
<pre class="r"><code>common_SNP &lt;- find_common_snps(UKBB, GLGC)
UKBB_subset &lt;- UKBB %&gt;%
    filter(CHR_POS %in% common_SNP) %&gt;%
    arrange(match(CHR_POS, common_SNP))
GLGC_subset &lt;- GLGC %&gt;%
    filter(CHR_POS %in% common_SNP) %&gt;%
    arrange(match(CHR_POS, common_SNP))</code></pre>
<ul>
<li><strong>Identify Variation &amp; Adjust BETA &amp; Z Score:</strong>
Identify variation in reference alleles across ancestries, and adjust
the sign of the BETA and Z score.</li>
</ul>
<p>In this step, we employ the <code>allele_flip</code> function to
align the reference alleles of the GLGC dataset with those of the UKBB
dataset. If there are discrepancies in the reference alleles, the beta
effect size is adjusted accordingly.</p>
<pre class="r"><code>GLGC_subset_flip &lt;- allele_flip(UKBB_subset, GLGC_subset)</code></pre>
</div>
<div id="reference-panel-prepation" class="section level3">
<h3>2. Reference Panel Prepation:</h3>
<ul>
<li><strong>Subset SNPs:</strong> Subset to a shared group of SNPs
across ancestries and matched up with GWAS.</li>
</ul>
<pre class="r"><code># Read in the bim file of both ancestries
WB_bim &lt;- fread(&quot;/net/fantasia/home/borang/Susie_Mult/Revision_Round_1/QC_function_data/WB/chr_filter_22.bim&quot;)
BB_bim &lt;- fread(&quot;/net/fantasia/home/borang/Susie_Mult/Revision_Round_1/QC_function_data/BB/chr_filter_22.bim&quot;)
WB_bim &lt;- WB_bim %&gt;%
    rename(CHR = V1, POS = V4, REF = V6, ALT = V5) %&gt;%
    mutate(CHR_POS = paste0(CHR, &quot;_&quot;, POS))
BB_bim &lt;- BB_bim %&gt;%
    rename(CHR = V1, POS = V4, REF = V6, ALT = V5) %&gt;%
    mutate(CHR_POS = paste0(CHR, &quot;_&quot;, POS))

# Do QC
WB_bim &lt;- GWAS_QC(WB_bim, 0.001)  #Note GWAS_QC works if no MAF is provided
BB_bim &lt;- GWAS_QC(BB_bim, 0.001)

# Find a common set of the SNPs across ancestries
common_SNP_bim &lt;- find_common_snps(WB_bim, BB_bim)
WB_bim_subset &lt;- WB_bim %&gt;%
    filter(CHR_POS %in% common_SNP_bim)

# Find a common set of the SNPs with GWAS summary statistics
common_SNP_GWAS &lt;- find_common_snps(UKBB_subset, WB_bim_subset)
UKBB_used &lt;- UKBB_subset %&gt;%
    filter(CHR_POS %in% common_SNP_GWAS) %&gt;%
    arrange(match(CHR_POS, common_SNP_GWAS))
GLGC_used &lt;- GLGC_subset_flip %&gt;%
    filter(CHR_POS %in% common_SNP_GWAS) %&gt;%
    arrange(match(CHR_POS, common_SNP_GWAS))</code></pre>
<ul>
<li><strong>Flip Alleles:</strong> Flip alleles if needed to achieve
consistency.</li>
</ul>
<pre class="r"><code>REF_Allele = data.frame(SNP = WB_bim_subset %&gt;%
    filter(CHR_POS %in% common_SNP) %&gt;%
    arrange(match(CHR_POS, common_SNP)) %&gt;%
    pull(V2), REF = UKBB_used %&gt;%
    pull(REF))
geno_dir &lt;- &quot;/net/fantasia/home/borang/Susie_Mult/Revision_Round_1/QC_function_data/&quot;
REF_Allele_name &lt;- paste0(geno_dir, &quot;WB/REF_Allele.txt&quot;)
write.table(REF_Allele, REF_Allele_name, col.names = F, row.names = F, quote = F)
system(paste0(&quot;~/software/plink2 --bfile &quot;, geno_dir, &quot;/WB/chr_filter_22 --ref-allele &quot;,
    REF_Allele_name, &quot; 2 1 --make-bed --out  &quot;, geno_dir, &quot;/WB/chr_used_22&quot;))
system(paste0(&quot;~/software/plink2 --bfile &quot;, geno_dir, &quot;/BB/chr_filter_22 --ref-allele &quot;,
    REF_Allele_name, &quot; 2 1 --make-bed --out  &quot;, geno_dir, &quot;/BB/chr_used_22&quot;))</code></pre>
</div>
<div id="check-for-ld-mismatches" class="section level3">
<h3>3. Check for LD Mismatches:</h3>
<p>We focus on the region from 21917190 to 22482774 bp to check the LD
mismatch</p>
<pre class="r"><code>min_POS = 21917190
max_POS = 22482774
UKBB_example &lt;- UKBB_used %&gt;%
    filter(POS &gt;= min_POS, POS &lt;= max_POS)
GLGC_example &lt;- GLGC_used %&gt;%
    filter(POS &gt;= min_POS, POS &lt;= max_POS)
SNPs_in_region &lt;- WB_bim_subset %&gt;%
    filter(CHR_POS %in% UKBB_example$CHR_POS) %&gt;%
    arrange(match(CHR_POS, UKBB_example$CHR_POS)) %&gt;%
    pull(V2)

create_correlation_matrix &lt;- function(geno_name, SNPs_in_region) {
    # Read in the genotype file selecting only the SNPs in the specified region
    geno_data &lt;- read.plink(paste0(geno_name, &quot;.bed&quot;), select.snps = SNPs_in_region)

    # Convert genotypes to numeric and match them with SNPs_in_region
    plink_geno &lt;- as(geno_data$genotypes, &quot;numeric&quot;)
    plink_geno &lt;- plink_geno[, match(SNPs_in_region, geno_data$map$snp.name)]

    # Replace NA values with the mean of the respective column (ignoring NA
    # values)
    plink_geno &lt;- apply(plink_geno, 2, function(x) {
        x[is.na(x)] &lt;- mean(x, na.rm = TRUE)
        return(x)
    })

    # Calculate the correlation matrix
    cov &lt;- cov2cor(crossprod(scale(plink_geno)))

    return(cov)
}

# Generate correlation matrices for WB and BB genotype datasets
WB_cov &lt;- create_correlation_matrix(paste0(geno_dir, &quot;WB/chr_used_22&quot;), SNPs_in_region)
BB_cov &lt;- create_correlation_matrix(paste0(geno_dir, &quot;BB/chr_used_22&quot;), SNPs_in_region)</code></pre>
<ul>
<li><strong>Use <code>kriging_rss</code> Function:</strong> We adapted
function of <code>kriging_rss</code> function from SuSiE for LD mismatch
variant detection</li>
</ul>
<pre class="r"><code>WB_diagnostic &lt;- kriging_rss(UKBB_example$Z, WB_cov)
BB_diagnostic &lt;- kriging_rss(GLGC_example$Z, BB_cov)</code></pre>
<ul>
<li><strong>Inspect LD Discrepancies:</strong> Inspect each ancestry for
LD discrepancies.</li>
</ul>
<pre class="r"><code>WB_diagnostic$plot</code></pre>
<p><img src="figure/GWAS_QC.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/GWAS_QC.Rmd/unnamed-chunk-8-1.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>BB_diagnostic$plot</code></pre>
<p><img src="figure/GWAS_QC.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-2">
Past versions of unnamed-chunk-8-2.png
</button>
</p>
<div id="fig-unnamed-chunk-8-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/GWAS_QC.Rmd/unnamed-chunk-8-2.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We see SNPs are more close to the line in UKBB and noisy in GLGC as
we use in-sample LD for UKBB, and external LD for GLGC.</p>
</div>
<div id="observations-recommendations" class="section level3">
<h3>4. Observations &amp; Recommendations:</h3>
<ul>
<li><p><strong>LD Mismatch Indicators:</strong> SNPs that deviate most
from the diagonal in the plots above suggest mismatches. It’s
recommended, per <a
href="https://stephenslab.github.io/susieR/">SuSiE</a> guidelines (Wang
et al., JRSSB 2020), to identify LD mismatched SNPs that meet these
criteria:</p>
<ul>
<li>A logLR test value greater than 2.</li>
<li>An absolute Z-value above 2.</li>
</ul></li>
<li><p><strong>Marginally Non-significant SNPs:</strong> In addition to
SuSiE guideline, we also found that SNPs with an absolute Z-value less
than 2. If they also:</p>
<ul>
<li>Have a abs(z_std_diff) greater than 3.</li>
<li>Show a high correlation (absolute r-value greater than 0.8) with
GWAS signals.</li>
</ul>
<p>They should be treated as LD mismatch SNPs. Such mismatches can
obstruct the IBSS algorithm’s functionality, causing the simultaneous
selection of a GWAS signal and its highly correlated non-signal.
Specifically, when the IBSS algorithm selects a GWAS signal, the
non-signal is assigned a significant effect, which can result in the
non-signal being chosen with an opposite effect in subsequent
iterations.</p></li>
<li><p><strong>Data Observations in UKBB V2:</strong> In the UKBB V2
dataset, we noticed mismatches where SNPs imputed using the 1000G panel
sometimes had incorrect genomic positions.</p></li>
<li><p><strong>Data Observations in Meta-analysis:</strong> The similar
pattern is observed when a SNP is highly correlated with a GWAS signal,
while the sample size as well as Z score of the SNP is extremely small
due to the platform difference between analysis.</p></li>
<li><p><strong>Recommendation:</strong> We urge users to employ the most
recent UKBB genotype datasets. Always check for LD mismatches before
initiating fine-mapping to ensure accurate results. Remove the
mismatched SNP from the analysis if needed. There are also alternative
software tools available for detecting LD mismatches, such as <a
href="https://github.com/mkanai/slalom">SLALOM</a> (Kanai et al., Cell
Genomics 2022), <a
href="https://doi.org/10.5281/zenodo.5516202">DENTIST</a> (Chen et al.,
NC 2021).</p></li>
</ul>
</div>
<div id="proof-of-concept" class="section level3">
<h3>5. Proof of concept:</h3>
<ul>
<li><strong>Create mismatch GWAS data:</strong> Set one of the GWAS
signal Z-score to 0.</li>
</ul>
<pre class="r"><code>which(abs(UKBB_example$Z) &gt; 10)</code></pre>
<pre><code> [1]  2  3  4  5  6  7  8  9 10 12 15 19 20 21 22 23 24 25 26 29 30 38 39 40 41
[26] 44 45 47 48 49 52 54 55 56 57 58 60 61 62 67 69 70 71 73 77 78 79 81 82 84
[51] 85 86 91 92 93 94 95 96 98 99</code></pre>
<pre class="r"><code>min(WB_cov[which(abs(UKBB_example$Z) &gt; 10), which(abs(UKBB_example$Z) &gt; 10)])</code></pre>
<pre><code>[1] 0.9181014</code></pre>
<pre class="r"><code>## We create a hypothetical GWAS by setting one of the SNP effect size being
## zero
UKBB_example_mismatch &lt;- UKBB_example
UKBB_example_mismatch$Z[2] = 0</code></pre>
<ul>
<li><strong>LD mismatch is detected:</strong></li>
</ul>
<pre class="r"><code>WB_diagnostic_mismatch &lt;- kriging_rss(UKBB_example_mismatch$Z, WB_cov)
WB_diagnostic_mismatch$plot</code></pre>
<p><img src="figure/GWAS_QC.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/GWAS_QC.Rmd/unnamed-chunk-10-1.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>WB_diagnostic_mismatch$conditional_dist[2, ]</code></pre>
<pre><code>  z  condmean   condvar z_std_diff logLR
2 0 -10.59557 0.1321804   29.14345     0</code></pre>
<p>Note logLR = 0, Z-score = 0. The kriging_rss of SuSiE will not report
this SNP.</p>
<ul>
<li><strong>SuSiE with and without LD mismatch:</strong></li>
</ul>
<pre class="r"><code>library(susieR)
susie_WB &lt;- susie_rss(UKBB_example$Z, WB_cov, check_prior = FALSE)
susie_WB$sets</code></pre>
<pre><code>$cs
$cs$L1
 [1]  2  3  4  5  6  7  8  9 10 12 15 19 20 21 22 23 24 25 26 29 30 38 39 40 41
[26] 45 47 48 49 52 54 55 56 57 58 73 77 78 81 84 85 86 91 92 93 94 95 96 98 99


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    0.9610531     0.9935668       0.9961523

$cs_index
[1] 1

$coverage
[1] 0.9554458

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>susie_WB_mismatch &lt;- susie_rss(UKBB_example_mismatch$Z, WB_cov, check_prior = FALSE)
susie_WB_mismatch$sets</code></pre>
<pre><code>$cs
$cs$L1
[1] 5

$cs$L2
[1] 2

$cs$L7
[1] 48 55


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    1.0000000     1.0000000       1.0000000
L2    1.0000000     1.0000000       1.0000000
L7    0.9999829     0.9999829       0.9999829

$cs_index
[1] 1 2 7

$coverage
[1] 1 1 1

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>save(UKBB_example, GLGC_example, WB_cov, BB_cov, file = &quot;/net/fantasia/home/borang/Susie_Mult/meSuSie_Analysis/data/MESuSiE_Example.RData&quot;)</code></pre>
<p>The credible set detected by SuSiE contains 50 highly correlated SNP,
which is expected. However, we see 3 different SNPs detected as causal
SNP when there is a LD mismatch.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] susieR_0.12.35    snpStats_1.46.0   Matrix_1.6-2      survival_3.5-7   
[5] dplyr_1.1.3       data.table_1.14.8 MESuSiE_1.0       workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] gtable_0.3.4             xfun_0.41                bslib_0.5.1             
 [4] ggplot2_3.4.4            processx_3.8.2           ggrepel_0.9.4           
 [7] lattice_0.22-5           callr_3.7.3              vctrs_0.6.4             
[10] tools_4.3.2              ps_1.7.5                 generics_0.1.3          
[13] parallel_4.3.2           tibble_3.2.1             fansi_1.0.5             
[16] highr_0.10               pkgconfig_2.0.3          lifecycle_1.0.4         
[19] farver_2.1.1             compiler_4.3.2           stringr_1.5.0           
[22] git2r_0.32.0             progress_1.2.2           munsell_0.5.0           
[25] getPass_0.2-2            httpuv_1.6.12            htmltools_0.5.7         
[28] sass_0.4.7               yaml_2.3.7               later_1.3.1             
[31] pillar_1.9.0             nloptr_2.0.3             crayon_1.5.2            
[34] jquerylib_0.1.4          whisker_0.4.1            tidyr_1.3.0             
[37] cachem_1.0.8             tidyselect_1.2.0         digest_0.6.33           
[40] stringi_1.8.1            purrr_1.0.2              labeling_0.4.3          
[43] RcppArmadillo_0.11.1.1.0 splines_4.3.2            cowplot_1.1.1           
[46] rprojroot_2.0.4          fastmap_1.1.1            grid_4.3.2              
[49] colorspace_2.1-0         cli_3.6.1                magrittr_2.0.3          
[52] Rfast_2.0.6              utf8_1.2.4               withr_2.5.2             
[55] prettyunits_1.2.0        scales_1.2.1             promises_1.2.1          
[58] RcppZiggurat_0.1.6       rmarkdown_2.25           httr_1.4.7              
[61] matrixStats_1.1.0        hms_1.1.3                evaluate_0.23           
[64] knitr_1.45               irlba_2.3.5.1            rlang_1.1.2             
[67] Rcpp_1.0.11              mixsqp_0.3-48            glue_1.6.2              
[70] formatR_1.14             BiocGenerics_0.48.1      reshape_0.8.9           
[73] rstudioapi_0.15.0        jsonlite_1.8.7           plyr_1.8.9              
[76] R6_2.5.1                 zlibbioc_1.48.0          fs_1.6.2                </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
