<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="borangao" />

<meta name="date" content="2022-11-09" />

<title>Installation and Motivating Example</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MESuSiE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="installation.html">Introduction</a>
</li>
<li>
  <a href="GWAS_QC.html">Data Preparation</a>
</li>
<li>
  <a href="Run_MESuSiE.html">Run MESuSiE</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis reproduce
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Simulation.html">Simulation</a>
    </li>
    <li>
      <a href="Lipid_711.html">711 Regions of Lipid</a>
    </li>
    <li>
      <a href="Lipid_810.html">810 Regions of Lipid</a>
    </li>
    <li>
      <a href="MCHC_SCZ.html">MCHC and SCZ</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/borangao/meSuSie_Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Installation and Motivating Example</h1>
<h4 class="author">borangao</h4>
<h4 class="date">2022-11-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-10-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>meSuSie_Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220530code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220530code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220530)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomborangaomeSuSieAnalysistreeba8db567b85c23b12dd18456c89353b3fcfc63a2targetblankba8db56a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/borangao/meSuSie_Analysis/tree/ba8db567b85c23b12dd18456c89353b3fcfc63a2" target="_blank">ba8db56</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomborangaomeSuSieAnalysistreeba8db567b85c23b12dd18456c89353b3fcfc63a2targetblankba8db56a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/borangao/meSuSie_Analysis/tree/ba8db567b85c23b12dd18456c89353b3fcfc63a2" target="_blank">ba8db56</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  data/GLGC_chr_22.txt
    Untracked:  data/MESuSiE_Example.RData
    Untracked:  data/UKBB_chr_22.txt

Unstaged changes:
    Deleted:    analysis/illustration.Rmd
    Deleted:    analysis/toy_example.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/installation.Rmd</code>) and HTML
(<code>docs/installation.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/ba8db567b85c23b12dd18456c89353b3fcfc63a2/analysis/installation.Rmd" target="_blank">ba8db56</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Update my analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/borangao/meSuSie_Analysis/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/installation.html" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/62ce4b3f045d8a95de6224cb4bca6e2890f28abc/analysis/installation.Rmd" target="_blank">62ce4b3</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Update my analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/borangao/meSuSie_Analysis/10cb2675fd78d87eba45962277bae630c1286202/docs/installation.html" target="_blank">10cb267</a>
</td>
<td>
borangao
</td>
<td>
2022-11-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/65fba54dc6812c6e5d0624bc67e8a30a14f55722/analysis/installation.Rmd" target="_blank">65fba54</a>
</td>
<td>
borangao
</td>
<td>
2022-11-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/borangao/meSuSie_Analysis/65fba54dc6812c6e5d0624bc67e8a30a14f55722/docs/installation.html" target="_blank">65fba54</a>
</td>
<td>
borangao
</td>
<td>
2022-11-09
</td>
<td>
Build site.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="installation-of-mesusie" class="section level1">
<h1>Installation of MESuSiE</h1>
<pre class="r"><code>library(devtools)
# Install MESuSiE
install_github(&quot;borangao/MESuSiE&quot;,dependencies = FALSE)
# Load MESuSiE
library(MESuSiE)</code></pre>
</div>
<div id="motivating-example" class="section level1">
<h1>Motivating Example</h1>
<p>The motivating example is based on a toy dataset outlined in the
manuscript.</p>
<div id="data-characteristics" class="section level2">
<h2>1. Data Characteristics</h2>
<p>The dataset contains:</p>
<ul>
<li>Six SNPs categorized into three distinct LD clusters.</li>
<li>Within each cluster, the SNPs show strong correlations.</li>
<li>Between clusters, SNP correlations are weak.</li>
</ul>
<div id="causal-snps" class="section level3">
<h3>Causal SNPs:</h3>
<p>The primary advantage of MESuSiE lies in its capacity to
differentiate between shared and ancestry-specific causal signals. For
demonstration purposes, we simulated a dataset containing one shared and
two ancestry-specific causal signals.</p>
<ul>
<li><strong>SNP 4</strong>: Simulated as a shared causal SNP across
ancestries.</li>
<li><strong>SNP 2</strong>: Simulated as the causal SNP specific to the
BB ancestry.</li>
<li><strong>SNP 6</strong>: Simulated as the causal SNP specific to the
WB ancestry.</li>
</ul>
</div>
<div id="snp-grouping" class="section level3">
<h3>SNP Grouping</h3>
<ol style="list-style-type: decimal">
<li><strong>Cluster 1</strong>: Contains SNP 1 and 2.</li>
<li><strong>Cluster 2</strong>: Contains SNP 3 and 4.</li>
<li><strong>Cluster 3</strong>: Contains SNP 5 and 6.</li>
</ol>
</div>
</div>
<div id="input-of-mesusie" class="section level2">
<h2>2. Input of MESuSiE</h2>
<p>The package comes with the necessary data included. To use MESuSiE,
two lists are required: one for summary statistics and another for LD
matrices, each from multiple ancestries. It’s essential to name each
element in these lists, ensuring that the naming is consistent between
the summary statistics and LD matrices.</p>
<pre class="r"><code>data(summ_stat_list)
data(LD_list)
summ_stat_list</code></pre>
<pre><code>$WB
           SNP         Beta          Se          Z     N POS
rs1890449 SNP1 -0.001806456 0.001825742 -0.9894367 3e+05   1
rs3122053 SNP2 -0.001749892 0.001825742 -0.9584555 3e+05   2
rs6600259 SNP3  0.019230827 0.001825742 10.5331576 3e+05   3
rs6681089 SNP4  0.019113531 0.001825742 10.4689120 3e+05   4
rs3008244 SNP5  0.018599221 0.001825742 10.1872128 3e+05   5
rs3008245 SNP6  0.018806821 0.001825742 10.3009200 3e+05   6

$BB
           SNP          Beta          Se          Z     N POS
rs1890449 SNP1  0.0146503149 0.001825742  8.0243079 3e+05   1
rs3122053 SNP2  0.0196888866 0.001825742 10.7840473 3e+05   2
rs6600259 SNP3  0.0176200508 0.001825742  9.6508993 3e+05   3
rs6681089 SNP4  0.0192921442 0.001825742 10.5667426 3e+05   4
rs3008244 SNP5 -0.0017014483 0.001825742 -0.9319216 3e+05   5
rs3008245 SNP6  0.0007568107 0.001825742  0.4145223 3e+05   6</code></pre>
<p>Each element in the summary statistics list represents a data frame
of GWAS summary statistics for each ancestry. This data frame should
have columns named SNP, Beta, Se, Z, and N, which correspond to the SNP
information, marginal effect size, standard error, Z-scores, and sample
size, respectively.</p>
<pre class="r"><code>LD_list</code></pre>
<pre><code>$WB
                  SNP1         SNP2         SNP3         SNP4          SNP5
rs1890449 1.0000000000 0.9985345076  0.008849574  0.008352736  0.0004252546
rs3122053 0.9985345076 1.0000000000  0.009187116  0.008718572  0.0006605135
rs6600259 0.0088495745 0.0091871156  1.000000000  0.992043794 -0.0238684132
rs6681089 0.0083527360 0.0087185716  0.992043794  1.000000000 -0.0263527244
rs3008244 0.0004252546 0.0006605135 -0.023868413 -0.026352724  1.0000000000
rs3008245 0.0005909073 0.0008813672 -0.023679608 -0.025942562  0.9961393233
                   SNP6
rs1890449  0.0005909073
rs3122053  0.0008813672
rs6600259 -0.0236796083
rs6681089 -0.0259425621
rs3008244  0.9961393233
rs3008245  1.0000000000

$BB
                  SNP1         SNP2        SNP3       SNP4        SNP5
rs1890449  1.000000000  0.627749083 0.006781938 0.01153312 -0.02141473
rs3122053  0.627749083  1.000000000 0.008521627 0.01231062 -0.01548369
rs6600259  0.006781938  0.008521627 1.000000000 0.92899225  0.01440080
rs6681089  0.011533117  0.012310618 0.928992254 1.00000000  0.01766183
rs3008244 -0.021414730 -0.015483693 0.014400801 0.01766183  1.00000000
rs3008245 -0.035179346 -0.019228441 0.012404065 0.01611964  0.79215823
                 SNP6
rs1890449 -0.03517935
rs3122053 -0.01922844
rs6600259  0.01240406
rs6681089  0.01611964
rs3008244  0.79215823
rs3008245  1.00000000</code></pre>
<p>Each element in the LD matrices list corresponds to an LD matrix for
a specific ancestry. The column names of this matrix should match the
SNP names found in the summary statistics, ensuring consistency across
data.</p>
<blockquote>
<p><strong>Note:</strong> For a detailed tutorial, please refer to <a
href="https://borangao.github.io/meSuSie_Analysis/GWAS_QC.html">Data
Preparation</a> and <a
href="https://borangao.github.io/meSuSie_Analysis/Run_MESuSiE.html">analysis</a>.</p>
</blockquote>
</div>
<div id="analysis" class="section level2">
<h2>3. Analysis</h2>
<div id="mesusie" class="section level3">
<h3>MESuSiE</h3>
<pre class="r"><code>MESuSiE_res&lt;-meSuSie_core(LD_list,summ_stat_list,L=10)</code></pre>
<pre><code>*************************************************************

  Multiple Ancestry Sum of Single Effect Model (MESuSiE)          

   Visit http://www.xzlab.org/software.html For Update            

            (C) 2022 Boran Gao, Xiang Zhou                        

              GNU General Public License                          

*************************************************************
# Start data processing for sufficient statistics 
# Create MESuSiE object 
# Start data analysis 

# Data analysis is done, and now generates result 

Potential causal SNPs with PIP &gt; 0.5:  SNP2 SNP4 SNP6 

Credible sets for effects: 
$cs
$cs$L1
[1] 4

$cs$L2
[1] 2

$cs$L3
[1] 5 6


$cs_category
     L1      L2      L3 
&quot;WB_BB&quot;    &quot;BB&quot;    &quot;WB&quot; 

$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    1.0000000     1.0000000       1.0000000
L2    1.0000000     1.0000000       1.0000000
L3    0.9961393     0.9980697       0.9980697

$cs_index
[1] 1 2 3

$coverage
[1] 0.9997947 1.0000000 1.0000000

$requested_coverage
[1] 0.95


 Use MESuSiE_Plot() for visualization
# Total time used for the analysis: 0 mins</code></pre>
<pre class="r"><code>MESuSiE_res$pip_config</code></pre>
<pre><code>             WB         BB      WB_BB
[1,] 0.07142857 0.07142857 0.02380952
[2,] 0.07142857 0.63214915 0.36785085
[3,] 0.07142857 0.07142857 0.02380952
[4,] 0.07142857 0.07142857 0.99979466
[5,] 0.18126816 0.07142857 0.05082666
[6,] 0.56691986 0.07142857 0.20098531</code></pre>
<p>We observed that three SNPs have a posterior inclusion probability
(PIP) exceeding the 0.5 threshold. Upon further examination of the PIP
for ancestry-specificity and shared traits, we identified that SNP4 is
shared, while SNP2 and SNP6 are ancestry-specific. The categories within
the credible set indicate the ancestries affected by the SNPs present in
the set.</p>
</div>
<div id="susie" class="section level3">
<h3>SuSiE</h3>
<pre class="r"><code>library(susieR)
susie_WB&lt;-susie_rss(summ_stat_list$WB$Z,LD_list$WB)
susie_BB&lt;-susie_rss(summ_stat_list$BB$Z,LD_list$BB)
susie_WB$pip</code></pre>
<pre><code>     SNP1      SNP2      SNP3      SNP4      SNP5      SNP6 
0.0000000 0.0000000 0.6052634 0.3947366 0.2398731 0.7601269 </code></pre>
<pre class="r"><code>susie_BB$pip</code></pre>
<pre><code>        SNP1         SNP2         SNP3         SNP4         SNP5         SNP6 
1.015177e-11 1.000000e+00 1.704749e-04 9.998295e-01 0.000000e+00 0.000000e+00 </code></pre>
<p>For the univariate SuSiE analysis, we found that SNP 3 and 6 are
signals distinct to Europeans. In contrast, SNP 2 and 4 are signals
unique to Africans, based on a PIP threshold of 0.5.</p>
</div>
<div id="paintor" class="section level3">
<h3>Paintor</h3>
<pre class="r"><code>##We load the Paintor result directly
paintor_res&lt;-read.table(&quot;/net/fantasia/home/borang/Susie_Mult/Revision_Round_1/Simulation/091223/data/toy_example/result/fig1.mcmc.paintor&quot;,header=T)
paintor_res$Posterior_Prob</code></pre>
<pre><code>[1] 0.040756 0.999980 0.987828 0.999956 0.993764 0.995080</code></pre>
<p>Paintor identifies SNP2-6 as signals without distinguishing
ancestry-specific or shared causal variant.</p>
</div>
</div>
<div id="visualization" class="section level2">
<h2>4. Visualization</h2>
<p><img src="figure/installation.Rmd/unnamed-chunk-7-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/installation.Rmd/unnamed-chunk-7-1.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggpubr_0.6.0    cowplot_1.1.1   dplyr_1.1.2     ggplot2_3.4.2  
[5] susieR_0.11.84  MESuSiE_1.0     devtools_2.4.3  usethis_2.2.1  
[9] workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0         farver_2.1.1             fastmap_1.1.1           
 [4] reshape_0.8.9            promises_1.2.0.1         digest_0.6.30           
 [7] lifecycle_1.0.3          ellipsis_0.3.2           processx_3.8.0          
[10] magrittr_2.0.3           compiler_4.3.1           rlang_1.1.1             
[13] sass_0.4.6               progress_1.2.2           tools_4.3.1             
[16] utf8_1.2.3               yaml_2.3.7               knitr_1.39              
[19] ggsignif_0.6.4           labeling_0.4.2           prettyunits_1.2.0       
[22] pkgbuild_1.4.2           curl_5.0.1               plyr_1.8.8              
[25] abind_1.4-5              pkgload_1.3.1            withr_2.5.1             
[28] purrr_1.0.1              grid_4.3.1               fansi_1.0.5             
[31] git2r_0.32.0             colorspace_2.1-0         scales_1.2.1            
[34] cli_3.6.1                rmarkdown_2.22           crayon_1.5.2            
[37] generics_0.1.3           remotes_2.4.2            rstudioapi_0.14         
[40] httr_1.4.6               RcppArmadillo_0.11.1.1.0 sessioninfo_1.2.2       
[43] cachem_1.0.8             stringr_1.5.0            parallel_4.3.1          
[46] matrixStats_1.0.0        vctrs_0.6.2              Matrix_1.5-4.1          
[49] jsonlite_1.8.3           carData_3.0-5            car_3.1-2               
[52] callr_3.7.3              hms_1.1.2                mixsqp_0.3-48           
[55] ggrepel_0.9.1            rstatix_0.7.2            irlba_2.3.5.1           
[58] jquerylib_0.1.4          tidyr_1.3.0              glue_1.6.2              
[61] nloptr_2.0.3             ps_1.7.2                 stringi_1.7.12          
[64] gtable_0.3.1             later_1.3.1              RcppZiggurat_0.1.6      
[67] munsell_0.5.0            tibble_3.2.1             pillar_1.9.0            
[70] htmltools_0.5.5          R6_2.5.1                 rprojroot_2.0.3         
[73] evaluate_0.18            lattice_0.20-45          highr_0.10              
[76] backports_1.4.1          Rfast_2.0.6              memoise_2.0.1           
[79] broom_1.0.5              httpuv_1.6.11            bslib_0.5.0             
[82] Rcpp_1.0.11              gridExtra_2.3            whisker_0.4.1           
[85] xfun_0.39                fs_1.6.2                 getPass_0.2-2           
[88] pkgconfig_2.0.3         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
