<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="borangao" />

<meta name="date" content="2023-10-09" />

<title>Lipid 711</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MESuSiE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="installation.html">Introduction</a>
</li>
<li>
  <a href="GWAS_QC.html">Data Preparation</a>
</li>
<li>
  <a href="Run_MESuSiE.html">Run MESuSiE</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis reproduce
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Simulation.html">Simulation</a>
    </li>
    <li>
      <a href="Lipid_711.html">711 Regions of Lipid</a>
    </li>
    <li>
      <a href="Lipid_810.html">810 Regions of Lipid</a>
    </li>
    <li>
      <a href="MCHC_SCZ.html">MCHC and SCZ</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/borangao/meSuSie_Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Lipid 711</h1>
<h4 class="author">borangao</h4>
<h4 class="date">2023-10-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-10-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>meSuSie_Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220530code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220530code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220530)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomborangaomeSuSieAnalysistree1f54cb830856dc649f6fe810b2e577ab43c53ecatargetblank1f54cb8a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/borangao/meSuSie_Analysis/tree/1f54cb830856dc649f6fe810b2e577ab43c53eca" target="_blank">1f54cb8</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomborangaomeSuSieAnalysistree1f54cb830856dc649f6fe810b2e577ab43c53ecatargetblank1f54cb8a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/borangao/meSuSie_Analysis/tree/1f54cb830856dc649f6fe810b2e577ab43c53eca" target="_blank">1f54cb8</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  data/GLGC_chr_22.txt
    Untracked:  data/MESuSiE_Example.RData
    Untracked:  data/UKBB_chr_22.txt

Unstaged changes:
    Deleted:    analysis/illustration.Rmd
    Deleted:    analysis/toy_example.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Lipid_711.Rmd</code>) and HTML
(<code>docs/Lipid_711.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/1f54cb830856dc649f6fe810b2e577ab43c53eca/analysis/Lipid_711.Rmd" target="_blank">1f54cb8</a>
</td>
<td>
borangao
</td>
<td>
2023-10-11
</td>
<td>
Update my analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/borangao/meSuSie_Analysis/428bdf044282b852602a3441b6bd45c13d6fdc1f/docs/Lipid_711.html" target="_blank">428bdf0</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/ba8db567b85c23b12dd18456c89353b3fcfc63a2/analysis/Lipid_711.Rmd" target="_blank">ba8db56</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Update my analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/borangao/meSuSie_Analysis/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/Lipid_711.html" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/62ce4b3f045d8a95de6224cb4bca6e2890f28abc/analysis/Lipid_711.Rmd" target="_blank">62ce4b3</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Update my analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="regions-of-4-lipid-traits" class="section level2">
<h2>711 Regions of 4 lipid traits</h2>
<blockquote>
<p><strong>Note</strong>: all the code and analysis reproduced here can
be found in <a
href="https://doi.org/10.5281/zenodo.8411004">Repository</a></p>
</blockquote>
</div>
<div id="feature-of-95-credible-set" class="section level2">
<h2>1. Feature of 95% credible set</h2>
<div id="a.-set-size-and-eqtl-enrichment-of-95-credible-set"
class="section level3">
<h3>1a. Set size and eQTL enrichment of 95% credible set</h3>
<pre class="r"><code>library(ggpubr)
library(data.table)
library(dplyr)
library(tidyr)
library(ggplot2)
library(patchwork)
library(ggpmisc)
library(VennDiagram)
library(gridExtra)
library(ggbreak)
library(DescTools)
library(coin)
library(susieR)
library(ggrepel)
library(stringr)


load(&quot;/net/fantasia/home/borang/Susie_Mult/Revision_Round_1/01_06_Real_Data/summary_res/res.RData&quot;)
custom_theme &lt;- function() {
  theme(
    axis.text.x = element_text(size = 5),
    axis.text.y = element_text(size = 5),  
    axis.title.x = element_text(size = 7, face=&quot;bold&quot;),
    axis.title.y = element_text(size = 7, face=&quot;bold&quot;),
    strip.text.x = element_text(size = 5),
    strip.text.y = element_text(size = 5),
    strip.background = element_blank(),
    legend.text = element_text(size=7),
    legend.title = element_text(size=7, face=&quot;bold&quot;),
    plot.title = element_text(size=7, hjust = 0.5),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank(), 
    axis.line = element_line(color = &quot;black&quot;)
  )
}
################################################
#
#       Set Size/Z-score/eQTL 
#
#
###############################################
################################################
#
#       Set SiZe Part
#
###############################################     
###Median set size by Trait
all_sets_info&lt;-data.frame(res_all%&gt;%group_by(Trait,Region) %&gt;% summarise(across(c(&quot;MESuSiE_cs&quot;, &quot;SuSiE_cs&quot;,&quot;Paintor_cs&quot;), ~ sum(.x, na.rm = TRUE))))%&gt;%filter(MESuSiE_cs!=0, SuSiE_cs!=0, Paintor_cs!=0) ###Median Set Size across all locus
all_sets_info_long&lt;-all_sets_info%&gt;%pivot_longer(!(Trait|Region), names_to = &quot;Method&quot;, values_to = &quot;Count&quot;)
all_sets_info_long$Method&lt;-factor(all_sets_info_long$Method,levels=c(&quot;MESuSiE_cs&quot;,&quot;SuSiE_cs&quot;,&quot;Paintor_cs&quot;))
levels(all_sets_info_long$Method)&lt;-c(&quot;MESuSiE&quot;,&quot;SuSiE&quot;,&quot;Paintor&quot;)

p_set = ggplot(data =all_sets_info_long,aes(x = Trait, y=Count,fill=Method))+geom_boxplot(aes(x = Trait,fill=Method),outlier.size = 0.1,fatten = 0.5,color = &quot;darkgray&quot;)+scale_fill_manual(values=c(&quot;MESuSiE&quot;=&quot;#023e8a&quot;,&quot;SuSiE&quot;=&quot;#2a9d8f&quot;,&quot;Paintor&quot;=&quot;#f4a261&quot;),guide=FALSE)
p_set =p_set + theme_bw() + xlab(&quot;&quot;) +ylab(&quot;Set Size&quot;)+coord_cartesian(ylim=c(0,175))
p_set= p_set+custom_theme()

################################################
#
#       Z-score Part
#
###############################################     
MESuSiE_cs_Z&lt;-res_all%&gt;%group_by(Trait) %&gt;%filter(MESuSiE_cs==1)%&gt;%summarise(zmax = median(pmax(abs(zscore_WB),abs(zscore_BB))))
SuSiE_cs_Z&lt;-res_all%&gt;%group_by(Trait) %&gt;%filter(SuSiE_cs==1)%&gt;%summarise(zmax =median(pmax(abs(zscore_WB),abs(zscore_BB))))%&gt;%pull(zmax)
Paintor_cs_Z&lt;-res_all%&gt;%group_by(Trait) %&gt;%filter(Paintor_cs==1)%&gt;%summarise(zmax = median(pmax(abs(zscore_WB),abs(zscore_BB))))%&gt;%pull(zmax)
set_size_z_info&lt;-data.frame(cbind(MESuSiE_cs_Z,SuSiE_cs_Z,Paintor_cs_Z))
colnames(set_size_z_info)&lt;-c(&quot;Trait&quot;,c(&quot;MESuSiE&quot;,&quot;SuSiE&quot;,&quot;Paintor&quot;))
set_size_z_info_long&lt;-set_size_z_info %&gt;%pivot_longer(!(Trait), names_to = &quot;Method&quot;, values_to = &quot;Z&quot;)%&gt;%mutate(Method = factor(Method, levels=c(&quot;MESuSiE&quot;,&quot;SuSiE&quot;,&quot;Paintor&quot;)))

p_z = ggplot(data = set_size_z_info_long,aes(x = Trait, y=Z,fill=Method))+geom_bar( stat = &quot;identity&quot;,position=&quot;dodge&quot;)+scale_fill_manual(values=c(&quot;MESuSiE&quot;=&quot;#023e8a&quot;,&quot;SuSiE&quot;=&quot;#2a9d8f&quot;,&quot;Paintor&quot;=&quot;#f4a261&quot;))
p_z = p_z + geom_text(label = round(set_size_z_info_long$Z,2),position = position_dodge(width = 1),vjust=-0.5,size = 5*5/14)
p_z = p_z + theme_bw() + xlab(&quot;&quot;) +ylab(&quot;Median |Z|&quot;)+ ylim(0,max(round(set_size_z_info_long$Z,2)+1))
p_z = p_z +custom_theme()
################################################
#
#       eQTL enrichment 
#
#
############################################### 
 ann_col_name&lt;-c(&quot;missense&quot;, &quot;synonymous&quot;, &quot;utr_comb&quot;, &quot;promotor&quot;, &quot;CRE&quot;,&quot;liver_ind_eQTL&quot;)
  # Functions for calculating fold enrichment
  calc_fold_enrichment &lt;- function(df, cs_col, ann_col_name) {
    df %&gt;%
      group_by(Region) %&gt;%
      filter(sum(!!sym(cs_col)) != 0) %&gt;%
      group_by(Trait, !!sym(cs_col)) %&gt;%
      summarise(across(ann_col_name, ~ sum(.x, na.rm = TRUE) / n())) %&gt;%
      group_by(Trait) %&gt;%
      summarise(across(ann_col_name, ~ .x[!!sym(cs_col) == 1] / .x[!!sym(cs_col) == 0]))
  }
  
  MESuSiE_PIP_ann &lt;- calc_fold_enrichment(res_all, &quot;MESuSiE_cs&quot;, ann_col_name)
  SuSiE_PIP_ann &lt;- calc_fold_enrichment(res_all, &quot;SuSiE_cs&quot;, ann_col_name)
  Paintor_PIP_ann &lt;- calc_fold_enrichment(res_all, &quot;Paintor_cs&quot;, ann_col_name)
  # Combine results
  Trait_CS_enrichment &lt;- bind_rows(
    MESuSiE_PIP_ann %&gt;% mutate(Method = &quot;MESuSiE&quot;),
    SuSiE_PIP_ann %&gt;% mutate(Method = &quot;SuSiE&quot;),
    Paintor_PIP_ann %&gt;% mutate(Method = &quot;Paintor&quot;)
  ) %&gt;% mutate(Method = factor(Method, levels = c(&quot;MESuSiE&quot;, &quot;SuSiE&quot;, &quot;Paintor&quot;)))%&gt;%
    dplyr::select(Trait,liver_ind_eQTL ,Method )%&gt;%dplyr::rename(eQTL = liver_ind_eQTL)
  # Pivot to long format
  Trait_CS_enrichment_long &lt;- Trait_CS_enrichment %&gt;%
    pivot_longer(cols = -c(Method, Trait), names_to = &quot;Cat&quot;, values_to = &quot;Prop&quot;) %&gt;%
    mutate(Method = factor(Method, levels = c(&quot;MESuSiE&quot;, &quot;SuSiE&quot;, &quot;Paintor&quot;)))

p_eQTL &lt;- ggplot(Trait_CS_enrichment_long, aes(x = Trait, y = Prop, fill = Method)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;dodge&quot;) +scale_fill_manual(values = c(&quot;MESuSiE&quot; = &quot;#023e8a&quot;, &quot;SuSiE&quot; = &quot;#2a9d8f&quot;, &quot;Paintor&quot; = &quot;#f4a261&quot;)) +
  geom_text(,label = round(Trait_CS_enrichment_long$Prop,2),position = position_dodge(width = 1),vjust=-0.5,size = 5*5/14)+
  xlab(&quot;&quot;) + ylab(&quot;eQTL Fold Enrichment&quot;) + ylim(0,max(round(Trait_CS_enrichment_long$Prop))+1)+
  theme_bw() + custom_theme()

p_out&lt;-p_set/p_z/p_eQTL+plot_annotation(tag_levels = &#39;a&#39;)+plot_layout(guides = &quot;collect&quot;,heights = c(1.5,1,1))&amp;theme(legend.position = &#39;bottom&#39;,plot.tag = element_text(size = 7,face=&quot;bold&quot;))
p_out</code></pre>
<p><img src="figure/Lipid_711.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/Lipid_711.Rmd/unnamed-chunk-1-1.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="b.-functional-enrichment-of-95-credible-set-and-top-signals"
class="section level3">
<h3>1b. Functional enrichment of 95% credible set and top signals</h3>
<pre class="r"><code>################################################################################
#
#
#     Functional Annotation enrichment for 95% credible set SNPS
#
#
################################################################################
 # Enrichment of 95% credible set without by trait
  calc_fold_enrichment_marginal&lt;-function(df, cs_col, ann_col_name) {
    df %&gt;%group_by(Region) %&gt;%
      filter(sum(!!sym(cs_col)) != 0) %&gt;%
      group_by( !!sym(cs_col)) %&gt;%
      summarise(across(ann_col_name, ~ sum(.x, na.rm = TRUE) / n())) %&gt;%
      summarise(across(ann_col_name, ~ .x[!!sym(cs_col) == 1] / .x[!!sym(cs_col) == 0]))
  }
  MESuSiE_PIP_ann &lt;- calc_fold_enrichment_marginal(res_all, &quot;MESuSiE_cs&quot;, ann_col_name)
  SuSiE_PIP_ann &lt;- calc_fold_enrichment_marginal(res_all, &quot;SuSiE_cs&quot;, ann_col_name)
  Paintor_PIP_ann &lt;- calc_fold_enrichment_marginal(res_all, &quot;Paintor_cs&quot;, ann_col_name)
  
  CS_enrichment &lt;- bind_rows(
    MESuSiE_PIP_ann %&gt;% mutate(Method = &quot;MESuSiE&quot;),
    SuSiE_PIP_ann %&gt;% mutate(Method = &quot;SuSiE&quot;),
    Paintor_PIP_ann %&gt;% mutate(Method = &quot;Paintor&quot;)
  ) %&gt;% mutate(Method = factor(Method, levels = c(&quot;MESuSiE&quot;, &quot;SuSiE&quot;, &quot;Paintor&quot;)))%&gt;% 
    dplyr::rename(Missense = missense ,Synonymous = synonymous,UTR = utr_comb,Promotor = promotor,eQTL = liver_ind_eQTL)
  
  # Pivot to long format
  CS_enrichment_long &lt;- CS_enrichment %&gt;%
    pivot_longer(cols = -c(Method), names_to = &quot;Cat&quot;, values_to = &quot;Prop&quot;) %&gt;%
    mutate(Method = factor(Method, levels = c(&quot;MESuSiE&quot;, &quot;SuSiE&quot;, &quot;Paintor&quot;))) %&gt;%
    mutate(Cat = factor(Cat, levels = c(&quot;Missense&quot;, &quot;Synonymous&quot;, &quot;UTR&quot;, &quot;Promotor&quot;, &quot;CRE&quot;,&quot;eQTL&quot;)))%&gt;%
    mutate(Prop = round(Prop, 2))

p_set &lt;- ggplot(data = CS_enrichment_long,aes(x = Cat, y = Prop, fill = Method)) +
  geom_col(position = &quot;dodge&quot;) + scale_fill_manual(values = c(&quot;MESuSiE&quot; = &quot;#023e8a&quot;, &quot;SuSiE&quot; = &quot;#2a9d8f&quot;, &quot;Paintor&quot; = &quot;#f4a261&quot;)) +
  geom_text(aes(x=Cat,group=Method,y=Prop,label=Prop),position = position_dodge(width = 1),vjust=-0.5,size = 5/14*5) + 
  geom_hline(yintercept = 1, linetype = &quot;dashed&quot;) + 
  xlab(&quot;&quot;) + ylab(&quot;Fold Enrichment Credible Set&quot;) +ylim(0,round(max(CS_enrichment_long$Prop))+1) +
  theme_bw() + custom_theme()
################################################################################
#
#
#     Functional Annotation enrichment for top 500 PIP SNPs
#
#
################################################################################    
  # Enrichment of top 500 signal
  res_all&lt;-res_all%&gt;%mutate(Paintor_Signal = ifelse(Paintor_PIP&gt;0.5,1,0))
  res_all&lt;-res_all%&gt;%mutate(SuSiE_Signal = case_when(
    SuSiE_WB&gt;0.5&amp;SuSiE_BB&gt;0.5~1,
    SuSiE_WB&gt;0.5&amp;SuSiE_BB&lt;0.5~2,
    SuSiE_WB&lt;0.5&amp;SuSiE_BB&gt;0.5~3,
    .default =0))
  res_all&lt;-res_all%&gt;%mutate(MESuSiE_Signal = case_when(
    MESuSiE_PIP_Shared&gt;0.5~1,
    MESuSiE_PIP_WB&gt;0.5~2,
    MESuSiE_PIP_BB&gt;0.5~3,
    .default =0))
  top_N_signal = 500
  bg_an&lt;-res_all%&gt;%summarise(across(ann_col_name,~ sum(.x, na.rm = TRUE)/n()))
  MESuSiE_Signal_ann&lt;-res_all%&gt;%filter(MESuSiE_Signal!=0)%&gt;% arrange(desc(MESuSiE_PIP_Either))%&gt;%top_n(n = top_N_signal, wt = MESuSiE_PIP_Either)%&gt;%summarise(across(ann_col_name,~ sum(.x, na.rm = TRUE)/n()))/bg_an
  SuSiE_Signal_ann&lt;-res_all%&gt;%filter(SuSiE_Signal!=0)%&gt;% arrange(desc(SuSiE_PIP))%&gt;%top_n(n = top_N_signal, wt = SuSiE_PIP)%&gt;%summarise(across(ann_col_name,~ sum(.x, na.rm = TRUE)/n()))/bg_an
  Paintor_Signal_ann&lt;-res_all%&gt;%filter(Paintor_Signal!=0) %&gt;% arrange(desc(Paintor_PIP))%&gt;%top_n(n = top_N_signal, wt = Paintor_PIP)%&gt;%summarise(across(ann_col_name,~ sum(.x, na.rm = TRUE)/n()))/bg_an
  
  Signal_enrichment &lt;- bind_rows(
    MESuSiE_Signal_ann %&gt;% mutate(Method = &quot;MESuSiE&quot;),
    SuSiE_Signal_ann %&gt;% mutate(Method = &quot;SuSiE&quot;),
    Paintor_Signal_ann %&gt;% mutate(Method = &quot;Paintor&quot;)
  ) %&gt;% mutate(Method = factor(Method, levels = c(&quot;MESuSiE&quot;, &quot;SuSiE&quot;, &quot;Paintor&quot;)))%&gt;% 
    dplyr::rename(Missense = missense ,Synonymous = synonymous,UTR = utr_comb,Promotor = promotor,eQTL = liver_ind_eQTL)
  # Pivot to long format
  Signal_enrichment_long &lt;- Signal_enrichment %&gt;%
    pivot_longer(cols = -c(Method), names_to = &quot;Cat&quot;, values_to = &quot;Prop&quot;) %&gt;%
    mutate(Method = factor(Method, levels = c(&quot;MESuSiE&quot;, &quot;SuSiE&quot;, &quot;Paintor&quot;))) %&gt;%
    mutate(Cat = factor(Cat, levels = c(&quot;Missense&quot;, &quot;Synonymous&quot;, &quot;UTR&quot;, &quot;Promotor&quot;, &quot;CRE&quot;,&quot;eQTL&quot;)))%&gt;%
    mutate(Prop = round(Prop, 2))

p_signal &lt;- ggplot(data = Signal_enrichment_long,aes(x = Cat, y = Prop, fill = Method)) +
  geom_col(position = &quot;dodge&quot;) + scale_fill_manual(values = c(&quot;MESuSiE&quot; = &quot;#023e8a&quot;, &quot;SuSiE&quot; = &quot;#2a9d8f&quot;, &quot;Paintor&quot; = &quot;#f4a261&quot;)) +
  geom_text(aes(x=Cat,group=Method,y=Prop,label=Prop),position = position_dodge(width = 1),vjust=-0.5,size = 5/14*5) + 
  geom_hline(yintercept = 1, linetype = &quot;dashed&quot;) + 
  xlab(&quot;&quot;) + ylab(&quot;Fold Enrichment Top Signal&quot;) +ylim(0,round(max(Signal_enrichment_long$Prop))+1) +
  theme_bw() + custom_theme()
p_out&lt;-p_set/p_signal+plot_annotation(tag_levels = &#39;a&#39;)+plot_layout(guides = &quot;collect&quot;,heights = c(1,1))&amp;theme(legend.position = &#39;bottom&#39;,plot.tag = element_text(size = 7,face=&quot;bold&quot;))
p_out</code></pre>
<p><img src="figure/Lipid_711.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/Lipid_711.Rmd/unnamed-chunk-2-1.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="top-signals-with-pip-0.5" class="section level2">
<h2>2. Top signals with PIP &gt; 0.5</h2>
<div id="a.-proportion-of-shared-and-ancestry-specific-signal"
class="section level3">
<h3>2a. Proportion of shared and ancestry-specific signal</h3>
<pre class="r"><code>############################################################################
#
#
#              Proportion of Signal Plot
#
#
############################################################################
Signal_number&lt;- res_all%&gt;%group_by(Trait)%&gt;%
  summarise(Paintor_Either_n = sum(Paintor_Signal!=0),
            SuSiE_Shared_n = sum(SuSiE_Signal==1),
            SuSiE_WB_n = sum(SuSiE_Signal==2),
            SuSiE_BB_n = sum(SuSiE_Signal==3),
            MESuSiE_Shared_n = sum(MESuSiE_Signal==1),
            MESuSiE_WB_n = sum(MESuSiE_Signal==2),
            MESuSiE_BB_n = sum(MESuSiE_Signal==3))
Signal_number&lt;-Signal_number%&gt;%
  pivot_longer(cols = -c(Trait), names_to = &quot;Cat&quot;, values_to = &quot;Num&quot;)%&gt;%
  separate(Cat, into = c(&quot;Method&quot;, &quot;Signal&quot;), sep = &quot;_&quot;, extra = &quot;merge&quot;) %&gt;%
  mutate(
    Method = case_when(
      str_detect(Method, &quot;MESuSiE&quot;) ~ &quot;MESuSiE&quot;,
      str_detect(Method, &quot;SuSiE&quot;) ~ &quot;SuSiE&quot;,
      str_detect(Method, &quot;Paintor&quot;) ~ &quot;Paintor&quot;,
      TRUE ~ Method
    ),
    Signal = case_when(
      str_detect(Signal, &quot;BB_n&quot;) ~ &quot;AFR&quot;,
      str_detect(Signal, &quot;WB_n&quot;) ~ &quot;EUR&quot;,
      str_detect(Signal, &quot;Shared_n&quot;) ~ &quot;Shared&quot;,
      str_detect(Signal, &quot;Either&quot;) ~ &quot;Either&quot;,
      TRUE ~ Signal
    )
  )
Signal_number&lt;-Signal_number%&gt;%group_by(Method,Trait)%&gt;%mutate(prop = Num/sum(Num)*100,ypos = cumsum(prop)- 0.5*prop)%&gt;%mutate(label = paste0(Signal,&quot; &quot;,Num))
Signal_number&lt;-Signal_number%&gt;%mutate(Method = factor(Method, levels = c(&quot;MESuSiE&quot;,&quot;SuSiE&quot;,&quot;Paintor&quot;)),Signal = factor(Signal, levels = c(&quot;EUR&quot;,&quot;AFR&quot;,&quot;Shared&quot;,&quot;Either&quot;)))

signal_num_plot&lt;-ggplot(Signal_number, aes(x=&quot;&quot;, y=prop, fill=Signal)) +
  geom_bar(stat=&quot;identity&quot;, width=1, color=&quot;white&quot;) +
  coord_polar(&quot;y&quot;, start=0) +
  theme_void() + 
  theme(legend.position=&quot;none&quot;) +
  geom_text(aes(y = ypos, label = label),  size=7/14*5,color=&quot;white&quot;) +
  scale_fill_manual(values=c(&quot;#ABDB9F&quot;,&quot;#F2C1B6&quot;,&quot;#6162B0&quot;,&quot;gray&quot;))+
  facet_grid(vars(Trait),vars(Method),labeller=label_parsed)+theme(
    strip.text.x = element_text(size = 7,face=&quot;bold&quot;),
    strip.text.y = element_text(size = 7,face=&quot;bold&quot;),
    strip.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank())
signal_num_plot</code></pre>
<p><img src="figure/Lipid_711.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/Lipid_711.Rmd/unnamed-chunk-3-1.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="b.-feature-of-shared-and-ancestry-specific-signal"
class="section level3">
<h3>2b. Feature of shared and ancestry-specific signal</h3>
<pre class="r"><code>###########################################################################################
#
#
#         Correlation of ancestry-specific signal by MESuSiE and SuSiE
#
#
#########################################################################################
cor_ancestry_MESuSiE &lt;- res_all %&gt;%
    filter(MESuSiE_Signal%in%c(2,3)) %&gt;% filter(!(SuSiE_Signal%in%c(2,3)))%&gt;%
    mutate(Method = &quot;MESuSiE&quot;) %&gt;%
    dplyr::select(Beta_WB, Beta_BB, Trait, Method)
  cor_ancestry_SuSiE &lt;- res_all %&gt;%
    filter(SuSiE_Signal%in%c(2,3)) %&gt;%filter(!(MESuSiE_Signal%in%c(2,3)))%&gt;%
    mutate(Method = &quot;SuSiE&quot;) %&gt;%
    dplyr::select(Beta_WB, Beta_BB, Trait, Method)
  cor_ancestry_both &lt;- res_all %&gt;%filter(MESuSiE_Signal%in%c(2,3),SuSiE_Signal%in%c(2,3))%&gt;%
    mutate(Method = &quot;Both&quot;) %&gt;%
    dplyr::select(Beta_WB, Beta_BB, Trait, Method)
  cor_ancestry &lt;- rbind(cor_ancestry_MESuSiE, cor_ancestry_SuSiE,cor_ancestry_both)
  cor_ancestry&lt;-cor_ancestry%&gt;%mutate(Method = factor(Method,levels = c(&quot;MESuSiE&quot;,&quot;SuSiE&quot;,&quot;Both&quot;))) 

betaplot&lt;- ggscatter(cor_ancestry, x = &quot;Beta_WB&quot;, y = &quot;Beta_BB&quot;,
                add = &quot;reg.line&quot;,  # Add regressin line
                add.params = list(color = &quot;red&quot;, fill = &quot;lightgray&quot;,size = 0.5), # Customize reg. line
                conf.int = FALSE, # Add confidence interval
                size = 0.5
)+ stat_cor(method = &quot;pearson&quot;,aes(label = ..r.label..),size = 5/14*5)+xlab(&quot;UKBB&quot;)+ylab(&quot;GLGC&quot;)+
  theme_bw() + custom_theme()+
  facet_wrap(vars(Method),ncol=3)
betaplot</code></pre>
<p><img src="figure/Lipid_711.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/Lipid_711.Rmd/unnamed-chunk-4-1.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#################################################################
#
#
#               MAF and Conservative Score Part
#
#
##################################################################
################################################
#
#       Phylop/GERP Score Part
#
#
################################################        

res_all&lt;- res_all %&gt;%
  mutate(
    MESuSiE_Conservative = case_when(
      MESuSiE_PIP_WB&gt;0.5|MESuSiE_PIP_BB&gt;0.5 ~ &quot;Ancestry-specific&quot;,
      MESuSiE_PIP_Shared&gt;0.5 ~ &quot;Shared&quot;,
      TRUE ~ &quot;None&quot;
    ),
    SuSiE_Conservative = case_when(
      (SuSiE_WB&gt;0.5&amp;SuSiE_BB&lt;0.5)|(SuSiE_BB&gt;0.5&amp;SuSiE_WB&lt;0.5) ~ &quot;Ancestry-specific&quot;,
      SuSiE_Shared&gt;0.5 ~ &quot;Shared&quot;,
      TRUE ~ &quot;None&quot;
    )
  )%&gt;%
  mutate(
    MESuSiE_Conservative = factor(MESuSiE_Conservative, levels = c(&quot;None&quot;, &quot;Ancestry-specific&quot;, &quot;Shared&quot;)),
    SuSiE_Conservative = factor(SuSiE_Conservative, levels = c(&quot;None&quot;, &quot;Ancestry-specific&quot;, &quot;Shared&quot;))
  )



####Phylop
grouped_data_phylop &lt;- res_all %&gt;%
  mutate(abs_phylop = abs(phylop)) %&gt;%
  select(MESuSiE_Conservative, abs_phylop)%&gt;%
  group_by(MESuSiE_Conservative) %&gt;%
  summarise(phylop_values = list(abs_phylop))%&gt;%
  pull(phylop_values)


grouped_data_GERP &lt;- res_all %&gt;%
  mutate(GERP = abs(GERP)) %&gt;%
  select(MESuSiE_Conservative, GERP)%&gt;%
  group_by(MESuSiE_Conservative) %&gt;%
  summarise(GERP_values = list(GERP))%&gt;%
  pull(GERP_values)


#########Conservation Plot
conserve_data&lt;-res_all%&gt;%mutate(Signal =MESuSiE_Conservative, Phylop =abs(phylop) )%&gt;%select(Signal,Phylop,GERP)
conserve_data_long&lt;-conserve_data%&gt;%pivot_longer(!(Signal), names_to = &quot;Method&quot;, values_to = &quot;Score&quot;)
conserve_data_long$Method&lt;-factor(conserve_data_long$Method,levels = c(&quot;Phylop&quot;,&quot;GERP&quot;))
df &lt;-conserve_data_long %&gt;%group_by(Method,Signal) %&gt;%summarize(ymin = min(Score,na.rm = T), lower = quantile(Score, .25,na.rm = T),  middle = mean(Score,na.rm = T), upper = quantile(Score, .75,na.rm = T),ymax = max(Score,na.rm = T)) 
p_set_phylop = df %&gt;% ggplot(aes(x = factor(Method), fill=Signal)) + geom_boxplot(fatten = 2,aes(ymin = ymin, ymax = ymax, lower = lower, upper = upper, middle = middle), stat = &#39;identity&#39;)+scale_fill_manual(values=c(&quot;None&quot;=&quot;#DAFFED&quot;,&quot;Ancestry-specific&quot;=&quot;#fffbdb&quot;,&quot;Shared&quot; = &quot;#7776bc&quot;))
p_set_phylop =p_set_phylop + theme_bw() + xlab(&quot;&quot;) +ylab(&quot;Conservation Score&quot;)
p_set_phylop= p_set_phylop  + custom_theme()
p_set_phylop= p_set_phylop+ theme(legend.position=&quot;bottom&quot;)     

################################################
#
#       MAF Part
#
#
################################################    
 maf_shared&lt;-data.frame(res_all)%&gt;%filter(MESuSiE_Signal==1)%&gt;%mutate(MAF_diff = ifelse(MA_WB==MA_BB,MAF_WB-MAF_BB,1-MAF_WB-MAF_BB))%&gt;%pull(MAF_diff)
  maf_ancestry&lt;-data.frame(res_all)%&gt;%filter(MESuSiE_Signal%in%c(2,3))%&gt;%mutate(MAF_diff = ifelse(MA_WB==MA_BB,MAF_WB-MAF_BB,1-MAF_WB-MAF_BB))%&gt;%pull(MAF_diff)
 

maf_data_plot&lt;-data.frame(&quot;MAF&quot; = c(maf_shared,maf_ancestry),&quot;Signal&quot;=c(rep(&quot;Shared&quot;,length(maf_shared)),rep(&quot;Ancestry-specific&quot;,length(maf_ancestry))))
maf_data_plot$Signal&lt;-factor(maf_data_plot$Signal)
maf_data_plot$Signal&lt;-factor(maf_data_plot$Signal,levels = c(levels(maf_data_plot$Signal),&quot;None&quot;))

p_set = ggplot(data =maf_data_plot,aes(x = factor(Signal), y=MAF,fill=Signal))+geom_boxplot(aes(x = factor(Signal),fill=Signal),outlier.size = 0.01) +scale_fill_manual(values=c(&quot;None&quot;=&quot;#DAFFED&quot;,&quot;Ancestry-specific&quot;=&quot;#fffbdb&quot;,&quot;Shared&quot; = &quot;#7776bc&quot;),guide=FALSE)
p_set = p_set + stat_compare_means(size= 5/14*5)
p_set =p_set + theme_bw() + xlab(&quot;&quot;) +ylab(&quot;MAF difference&quot;)
p_set= p_set + custom_theme()
p_set= p_set + theme(legend.position=&quot;none&quot;)    
p_out&lt;-p_set+p_set_phylop+plot_annotation(tag_levels = &#39;a&#39;) + plot_layout(guides = &quot;collect&quot;) &amp; theme(legend.position = &quot;bottom&quot;,plot.tag = element_text(size = 7,face=&quot;bold&quot;))
p_out</code></pre>
<p><img src="figure/Lipid_711.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/Lipid_711.Rmd/unnamed-chunk-4-2.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="example-locus" class="section level2">
<h2>3. Example locus</h2>
<div id="a.-fad2-example" class="section level3">
<h3>3a. FAD2 Example</h3>
<pre class="r"><code>##############################################################
#
#
#       Real Data Example Plotter
#
#
################################################################

gwas_plot_fun &lt;- function(data_plot, xlab_name, ylab_name, yintercept) {
  
  p_manhattan = ggplot() + geom_point(data = data_plot%&gt;%filter(Lead_SNP==0), aes(x = POS, y = PIP, color = r2), size = 1)
  p_manhattan = p_manhattan + geom_point(data = data_plot%&gt;%filter(Lead_SNP==1), aes(x = POS, y = PIP), size = 1.5, color = &quot;red&quot;) +
    geom_text_repel(data = data_plot%&gt;%filter(Lead_SNP==1), mapping = aes(x = POS, y = PIP, label = SNP), vjust = 1.2, size = 7/14*5, show.legend =FALSE) 
  p_manhattan = p_manhattan +
    scale_color_stepsn(
      colors = c(&quot;navy&quot;, &quot;lightskyblue&quot;, &quot;green&quot;, &quot;orange&quot;, &quot;red&quot;),
      breaks = seq(0.2, 0.8, by = 0.2),
      limits = c(0, 1),
      show.limits = TRUE,
      na.value = &#39;grey50&#39;,
      name = expression(R^2)
    )
  p_manhattan = p_manhattan +
    geom_hline(
      yintercept = yintercept,
      linetype = &quot;dashed&quot;,
      color = &quot;grey50&quot;,
      size = 0.5
    ) 
  p_manhattan = p_manhattan +
    geom_vline(
      xintercept = data_plot%&gt;%filter(lead_SNP==1)%&gt;%pull(POS),
      linetype = &quot;dashed&quot;,
      color = &quot;grey50&quot;,
      size = 0.5
    ) 
  p_manhattan = p_manhattan + xlim(min(data_plot$POS),max(data_plot$POS))
  p_manhattan = p_manhattan + expand_limits(x = round(max(data_plot$POS)/1e6)*1e6)
  if(max(data_plot$POS&gt;1e6)){
    p_manhattan = p_manhattan + scale_x_continuous(labels = function(x) paste0(x / 1e6, &quot; MB&quot;))
  }
  if(max(data_plot$POS&lt;1e6)){
    p_manhattan = p_manhattan + scale_x_continuous(labels = function(x) paste0(x / 1e3, &quot; KB&quot;))
  }
  p_manhattan = p_manhattan + xlab(xlab_name) +ylab(ylab_name)
  p_manhattan = p_manhattan + guides(fill = guide_legend(title = as.expression(bquote(R^2))))
  p_manhattan = p_manhattan + theme_bw()+custom_theme()
  return(p_manhattan)
}

###Function used for PIP plot   
finemap_plot_fun&lt;-function(data_plot,xlab_name,ylab_name,yintercept){
  p_manhattan = ggplot() + geom_point(data = data_plot, aes(x = POS, y = PIP, color = r2,shape = cat))+scale_shape_manual(name=&quot;Category&quot;,drop=FALSE,values=c(20,24,25,23,22))
  p_manhattan = p_manhattan + geom_text_repel(data =data_plot%&gt;%filter(Lead_SNP==1), mapping=aes(x=POS, y=PIP, label=SNP),vjust=1.2, size= 7/14*5,show.legend = FALSE)
  p_manhattan = p_manhattan + theme_bw()+scale_color_stepsn(
    colors = c(&quot;navy&quot;, &quot;lightskyblue&quot;, &quot;green&quot;, &quot;orange&quot;, &quot;red&quot;),
    breaks = seq(0.2, 0.8, by = 0.2),
    limits = c(0, 1),
    show.limits = TRUE,
    na.value = &#39;grey50&#39;,
    name = expression(R^2)
  )
  p_manhattan = p_manhattan + geom_hline(
    yintercept =yintercept,
    linetype = &quot;dashed&quot;,
    color = &quot;grey50&quot;,
    size = 0.5
  ) + geom_vline(
    xintercept = data_plot%&gt;%filter(lead_SNP==1)%&gt;%pull(POS),
    linetype = &quot;dashed&quot;,
    color = &quot;grey50&quot;,
    size = 0.5
  ) 
  p_manhattan = p_manhattan + xlim(min(data_plot$POS),max(data_plot$POS))
  p_manhattan = p_manhattan + expand_limits(x = round(max(data_plot$POS)/1e6)*1e6)
  if(max(data_plot$POS&gt;1e6)){
    p_manhattan = p_manhattan + scale_x_continuous(labels = function(x) paste0(x / 1e6, &quot; MB&quot;))
  }
  if(max(data_plot$POS&lt;1e6)){
    p_manhattan = p_manhattan + scale_x_continuous(labels = function(x) paste0(x / 1e3, &quot; KB&quot;))
  }
  p_manhattan= p_manhattan+xlab(xlab_name)+ylab(ylab_name)
  p_manhattan= p_manhattan+guides(fill=guide_legend(title=as.expression(bquote(R^2))))
  p_manhattan = p_manhattan + theme_bw()+custom_theme()
  return(p_manhattan)
}           
# Function used for gene plot 
gene_range_plot_fun&lt;-function(gene_list_data,plot.range){
  p&lt;-ggplot(data = gene_list_data) +
    geom_linerange(aes(x = Gene, ymin = Start, ymax = End))+ylim(plot.range)+ expand_limits(y = round(max(plot.range[2])/1e6)*1e6)+scale_y_continuous(labels = function(y) paste0(y / 1e6, &quot; MB&quot;))+coord_flip()+
    geom_text(aes(x = Gene, y = Start, label = Gene), hjust = &quot;right&quot;, size = 5/14*5) + ylab(paste0(&quot;chr&quot;,unique(gsub(&quot;chr&quot;,&quot;&quot;,gene_list_data$Chrom))))+ xlab(&quot;Gene&quot;) + 
    theme_bw() +  theme(
      axis.text.x = element_text(size = 5),
      axis.text.y = element_blank(),  
      axis.ticks.y =  element_blank(), 
      axis.title.x = element_text(size = 7, face=&quot;bold&quot;),
      axis.title.y = element_text(size = 7, face=&quot;bold&quot;),
      strip.text.x = element_text(size = 5),
      strip.text.y = element_text(size = 5),
      strip.background = element_blank(),
      legend.text = element_text(size=7),
      legend.title = element_text(size=7, face=&quot;bold&quot;),
      plot.title = element_text(size=7, hjust = 0.5),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank(), 
      axis.line.x  = element_line(color = &quot;black&quot;),
      axis.line.y  = element_line(color = &quot;black&quot;)
    )
  return(p)
}

#####################################################################################################################
#
#
#
#                 Example showcase
#
#
#
#####################################################################################################################
Gene_List&lt;-fread(&quot;/net/fantasia/home/borang/Susie_Mult/simulation/simu_0120/data/Gencode_GRCh37_Genes_UniqueList2021.txt&quot;,header=T)

###################################################################
#
#
#             FADS2 independent eQTL shared signal
#
#
##################################################################
region = 384
trait_name = &quot;TG&quot;
ref_panel = &quot;UKB1&quot;
p_threshold_index = 1
res_z_dir&lt;-paste0(&quot;/net/fantasia/home/borang/Susie_Mult/Revision_Round_1/01_02_Real_Data/&quot;,trait_name,&quot;/&quot;,trait_name,&quot;_REF_&quot;,ref_panel,&quot;_P&quot;,p_threshold_index,&quot;/&quot;)
out_dir&lt;-paste0(res_z_dir,&quot;data/&quot;)
out_res_dir&lt;-paste0(res_z_dir,&quot;res/&quot;)

##Data Reprocess
WB_COV&lt;-as.matrix(fread(paste0(out_dir,&quot;Region_&quot;,region,&quot;.LD1&quot;)))
BB_COV&lt;-as.matrix(fread(paste0(out_dir,&quot;Region_&quot;,region,&quot;.LD2&quot;)))
load(paste0(out_res_dir,&quot;MESuSiE_region_&quot;,region,&quot;.RData&quot;))

candidate_region&lt;-res_all%&gt;%filter(Region==region)
# rs174564 is the eQTL of FADS2 gene, highlighted in the paper
lead_SNP = &quot;rs174564&quot;
lead_SNP_index&lt;-which(candidate_region$SNP==lead_SNP)
candidate_region&lt;-candidate_region%&gt;%mutate(r2_EUR  = unname(unlist((WB_COV[,lead_SNP_index])^2)) ,r2_AFR = unname(unlist((BB_COV[,lead_SNP_index])^2)),POS = as.numeric(POS))
####Category Setting
candidate_region&lt;-candidate_region%&gt;%mutate(SuSiE_cat = case_when(SuSiE_WB&gt;0.5&amp;SuSiE_BB&gt;0.5 ~ 3,
                                                                  SuSiE_WB&gt;0.5&amp;SuSiE_BB&lt;0.5 ~ 1,
                                                                  SuSiE_WB&lt;0.5&amp;SuSiE_BB&gt;0.5 ~ 2,
                                                                  TRUE ~ 0),
                                            Paintor_cat = case_when(Paintor_PIP&gt;0.5~4,
                                                                    TRUE~0),
                                            MESuSiE_cat = case_when(MESuSiE_PIP_WB&gt;0.5~1,
                                                                    MESuSiE_PIP_BB&gt;0.5~2,
                                                                    MESuSiE_PIP_Shared&gt;0.5~3,
                                                                    TRUE~0))


###GWAS PLOT
EUR_GWAS_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_EUR,PIP = -log10(2*pnorm(-abs(zscore_WB))),Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP)
AFR_GWAS_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_AFR,PIP = -log10(2*pnorm(-abs(zscore_BB))),Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP)
p_EUR&lt;-gwas_plot_fun (EUR_GWAS_plot_data, &quot;UKBB&quot;, &quot;-log10(P-value)&quot;, -log10(5e-8))
p_AFR&lt;-gwas_plot_fun (AFR_GWAS_plot_data, &quot;GLGC&quot;, &quot;-log10(P-value)&quot;, -log10(5e-8))

###Finemap Plot
EUR_SuSiE_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_EUR,PIP = SuSiE_WB,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(SuSiE_cat,levels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;), labels = c(&quot;Non&quot;, &quot;EUR&quot;, &quot;AFR&quot;, &quot;Shared&quot;, &quot;Paintor&quot;)))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP,cat)
AFR_SuSiE_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_AFR,PIP = SuSiE_BB,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(SuSiE_cat,levels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;), labels = c(&quot;Non&quot;, &quot;EUR&quot;, &quot;AFR&quot;, &quot;Shared&quot;, &quot;Paintor&quot;)))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP,cat)
MESuSiE_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_EUR,PIP = MESuSiE_PIP_Either,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(MESuSiE_cat,levels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;), labels = c(&quot;Non&quot;, &quot;EUR&quot;, &quot;AFR&quot;, &quot;Shared&quot;, &quot;Paintor&quot;)))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP,cat)
Paintor_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_EUR,PIP = Paintor_PIP,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(Paintor_cat,levels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;), labels = c(&quot;Non&quot;, &quot;EUR&quot;, &quot;AFR&quot;, &quot;Shared&quot;, &quot;Paintor&quot;)))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP,cat)

p_EUR_SuSiE&lt;-finemap_plot_fun(EUR_SuSiE_plot_data, &quot;SuSiE UKBB&quot;, &quot;PIP&quot;, 0.5)
p_AFR_SuSiE&lt;-finemap_plot_fun(AFR_SuSiE_plot_data, &quot;SuSiE GLGC&quot;, &quot;PIP&quot;, 0.5)
p_MESuSiE&lt;-finemap_plot_fun(MESuSiE_plot_data, &quot;MESuSiE&quot;, &quot;PIP&quot;, 0.5)
p_Paintor&lt;-finemap_plot_fun(Paintor_plot_data, &quot;Paintor&quot;, &quot;PIP&quot;, 0.5)

# Gene Plot
plot.range &lt;- c(min(candidate_region$POS), max(candidate_region$POS))
Gene_List_sub_coding&lt;-Gene_List%&gt;%filter(Chrom==paste0(&quot;chr&quot;,unique(candidate_region$CHR)))%&gt;%filter(Start&lt;max(candidate_region$POS),End&gt;min(candidate_region$POS))%&gt;%filter(Coding==&quot;proteincoding&quot;)%&gt;%filter(!is.na(cdsLength))%&gt;%filter(GeneLength&gt;15000)
p2&lt;-gene_range_plot_fun(Gene_List_sub_coding,plot.range)

##Combine Plot together
combined_plot&lt;-(p_EUR/p_EUR_SuSiE/p_MESuSiE/p2+plot_layout(heights = c(1,1,1,1.5))|p_AFR/p_AFR_SuSiE/p_Paintor/p2+plot_layout(heights = c(1,1,1,1.5)))+plot_layout(guides = &#39;collect&#39;)&amp;theme(legend.position = &quot;bottom&quot;)
combined_plot</code></pre>
<p><img src="figure/Lipid_711.Rmd/unnamed-chunk-5-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/Lipid_711.Rmd/unnamed-chunk-5-1.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="b.-apoh-example" class="section level3">
<h3>3b. APOH Example</h3>
<pre class="r"><code>##############################################################
#
#
#           APOH missense shared for TG (rs1801689)
#
################################################################    
region = 438
trait_name = &quot;TG&quot;
ref_panel = &quot;UKB1&quot;
p_threshold_index = 1
res_z_dir&lt;-paste0(&quot;/net/fantasia/home/borang/Susie_Mult/Revision_Round_1/01_02_Real_Data/&quot;,trait_name,&quot;/&quot;,trait_name,&quot;_REF_&quot;,ref_panel,&quot;_P&quot;,p_threshold_index,&quot;/&quot;)
out_dir&lt;-paste0(res_z_dir,&quot;data/&quot;)
out_res_dir&lt;-paste0(res_z_dir,&quot;res/&quot;)

##Date Reprocess
WB_COV&lt;-as.matrix(fread(paste0(out_dir,&quot;Region_&quot;,region,&quot;.LD1&quot;)))
BB_COV&lt;-as.matrix(fread(paste0(out_dir,&quot;Region_&quot;,region,&quot;.LD2&quot;)))
load(paste0(out_res_dir,&quot;MESuSiE_region_&quot;,region,&quot;.RData&quot;))

candidate_region&lt;-res_all%&gt;%filter(Region==region)
# rs1801689 is the missense of APOH gene, highlighted in the paper
lead_SNP = &quot;rs1801689&quot;
lead_SNP_index&lt;-which(candidate_region$SNP==lead_SNP)
candidate_region&lt;-candidate_region%&gt;%mutate(r2_EUR  = unname(unlist((WB_COV[,lead_SNP_index])^2)) ,r2_AFR = unname(unlist((BB_COV[,lead_SNP_index])^2)),POS = as.numeric(POS))
####Category Setting
candidate_region&lt;-candidate_region%&gt;%mutate(SuSiE_cat = case_when(SuSiE_WB&gt;0.5&amp;SuSiE_BB&gt;0.5 ~ 3,
                                                                  SuSiE_WB&gt;0.5&amp;SuSiE_BB&lt;0.5 ~ 1,
                                                                  SuSiE_WB&lt;0.5&amp;SuSiE_BB&gt;0.5 ~ 2,
                                                                  TRUE ~ 0),
                                            Paintor_cat = case_when(Paintor_PIP&gt;0.5~4,
                                                                    TRUE~0),
                                            MESuSiE_cat = case_when(MESuSiE_PIP_WB&gt;0.5~1,
                                                                    MESuSiE_PIP_BB&gt;0.5~2,
                                                                    MESuSiE_PIP_Shared&gt;0.5~3,
                                                                    TRUE~0))

###GWAS PLOT
EUR_GWAS_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_EUR,PIP = -log10(2*pnorm(-abs(zscore_WB))),Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP)
AFR_GWAS_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_AFR,PIP = -log10(2*pnorm(-abs(zscore_BB))),Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP)
p_EUR&lt;-gwas_plot_fun (EUR_GWAS_plot_data, &quot;UKBB&quot;, &quot;-log10(P-value)&quot;, -log10(5e-8))
p_AFR&lt;-gwas_plot_fun (AFR_GWAS_plot_data, &quot;GLGC&quot;, &quot;-log10(P-value)&quot;, -log10(5e-8))

###Finemap Plot
EUR_SuSiE_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_EUR,PIP = SuSiE_WB,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(SuSiE_cat,levels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;), labels = c(&quot;Non&quot;, &quot;EUR&quot;, &quot;AFR&quot;, &quot;Shared&quot;, &quot;Paintor&quot;)))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP,cat)
AFR_SuSiE_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_AFR,PIP = SuSiE_BB,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(SuSiE_cat,levels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;), labels = c(&quot;Non&quot;, &quot;EUR&quot;, &quot;AFR&quot;, &quot;Shared&quot;, &quot;Paintor&quot;)))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP,cat)
MESuSiE_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_EUR,PIP = MESuSiE_PIP_Either,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(MESuSiE_cat,levels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;), labels = c(&quot;Non&quot;, &quot;EUR&quot;, &quot;AFR&quot;, &quot;Shared&quot;, &quot;Paintor&quot;)))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP,cat)
Paintor_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_EUR,PIP = Paintor_PIP,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(Paintor_cat,levels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;), labels = c(&quot;Non&quot;, &quot;EUR&quot;, &quot;AFR&quot;, &quot;Shared&quot;, &quot;Paintor&quot;)))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP,cat)

p_EUR_SuSiE&lt;-finemap_plot_fun(EUR_SuSiE_plot_data, &quot;SuSiE UKBB&quot;, &quot;PIP&quot;, 0.5)
p_AFR_SuSiE&lt;-finemap_plot_fun(AFR_SuSiE_plot_data, &quot;SuSiE GLGC&quot;, &quot;PIP&quot;, 0.5)
p_MESuSiE&lt;-finemap_plot_fun(MESuSiE_plot_data, &quot;MESuSiE&quot;, &quot;PIP&quot;, 0.5)
p_Paintor&lt;-finemap_plot_fun(Paintor_plot_data, &quot;Paintor&quot;, &quot;PIP&quot;, 0.5)

# Gene Plot
plot.range &lt;- c(min(candidate_region$POS), max(candidate_region$POS))
Gene_List_sub_coding&lt;-Gene_List%&gt;%filter(Chrom==paste0(&quot;chr&quot;,unique(candidate_region$CHR)))%&gt;%filter(Start&lt;max(candidate_region$POS),End&gt;min(candidate_region$POS))%&gt;%filter(Coding==&quot;proteincoding&quot;)%&gt;%filter(!is.na(cdsLength))%&gt;%filter(GeneLength&gt;15000)
p2&lt;-gene_range_plot_fun(Gene_List_sub_coding,plot.range)

##Combine Plot together
combined_plot&lt;-(p_EUR/p_EUR_SuSiE/p_MESuSiE/p2+plot_layout(heights = c(1,1,1,1.5))|p_AFR/p_AFR_SuSiE/p_Paintor/p2+plot_layout(heights = c(1,1,1,1.5)))+plot_layout(guides = &#39;collect&#39;)&amp;theme(legend.position = &quot;bottom&quot;)
combined_plot</code></pre>
<p><img src="figure/Lipid_711.Rmd/unnamed-chunk-6-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/Lipid_711.Rmd/unnamed-chunk-6-1.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="c.-tm6sf2-example" class="section level3">
<h3>3c. TM6SF2 Example</h3>
<pre class="r"><code>#######################################################################
#
#           TM6SF2 gene and LDL association with ancestry-specific effect
#
#
##########################################################################

region = 275
trait_name = &quot;LDL&quot;
ref_panel = &quot;UKB1&quot;
p_threshold_index = 1
res_z_dir&lt;-paste0(&quot;/net/fantasia/home/borang/Susie_Mult/Revision_Round_1/01_02_Real_Data/&quot;,trait_name,&quot;/&quot;,trait_name,&quot;_REF_&quot;,ref_panel,&quot;_P&quot;,p_threshold_index,&quot;/&quot;)
out_dir&lt;-paste0(res_z_dir,&quot;data/&quot;)
out_res_dir&lt;-paste0(res_z_dir,&quot;res/&quot;)

##Date Reprocess
WB_COV&lt;-as.matrix(fread(paste0(out_dir,&quot;Region_&quot;,region,&quot;.LD1&quot;)))
BB_COV&lt;-as.matrix(fread(paste0(out_dir,&quot;Region_&quot;,region,&quot;.LD2&quot;)))
load(paste0(out_res_dir,&quot;MESuSiE_region_&quot;,region,&quot;.RData&quot;))

candidate_region&lt;-res_all%&gt;%filter(Region==region)
# rs58542926 is the missense of TM6SF2 gene gene, highlighted in the paper
lead_SNP = &quot;rs58542926&quot;
lead_SNP_index&lt;-which(candidate_region$SNP==lead_SNP)
candidate_region&lt;-candidate_region%&gt;%mutate(r2_EUR  = unname(unlist((WB_COV[,lead_SNP_index])^2)) ,r2_AFR = unname(unlist((BB_COV[,lead_SNP_index])^2)),POS = as.numeric(POS))
####Category Setting
candidate_region&lt;-candidate_region%&gt;%mutate(SuSiE_cat = case_when(SuSiE_WB&gt;0.5&amp;SuSiE_BB&gt;0.5 ~ 3,
                                                                  SuSiE_WB&gt;0.5&amp;SuSiE_BB&lt;0.5 ~ 1,
                                                                  SuSiE_WB&lt;0.5&amp;SuSiE_BB&gt;0.5 ~ 2,
                                                                  TRUE ~ 0),
                                            Paintor_cat = case_when(Paintor_PIP&gt;0.5~4,
                                                                    TRUE~0),
                                            MESuSiE_cat = case_when(MESuSiE_PIP_WB&gt;0.5~1,
                                                                    MESuSiE_PIP_BB&gt;0.5~2,
                                                                    MESuSiE_PIP_Shared&gt;0.5~3,
                                                                    TRUE~0))
###GWAS PLOT
EUR_GWAS_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_EUR,PIP = -log10(2*pnorm(-abs(zscore_WB))),Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP)
AFR_GWAS_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_AFR,PIP = -log10(2*pnorm(-abs(zscore_BB))),Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP)
p_EUR&lt;-gwas_plot_fun (EUR_GWAS_plot_data, &quot;UKBB&quot;, &quot;-log10(P-value)&quot;, -log10(5e-8))
p_AFR&lt;-gwas_plot_fun (AFR_GWAS_plot_data, &quot;GLGC&quot;, &quot;-log10(P-value)&quot;, -log10(5e-8))

###Finemap Plot
EUR_SuSiE_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_EUR,PIP = SuSiE_WB,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(SuSiE_cat,levels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;), labels = c(&quot;Non&quot;, &quot;EUR&quot;, &quot;AFR&quot;, &quot;Shared&quot;, &quot;Paintor&quot;)))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP,cat)
AFR_SuSiE_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_AFR,PIP = SuSiE_BB,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(SuSiE_cat,levels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;), labels = c(&quot;Non&quot;, &quot;EUR&quot;, &quot;AFR&quot;, &quot;Shared&quot;, &quot;Paintor&quot;)))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP,cat)
MESuSiE_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_EUR,PIP = MESuSiE_PIP_Either,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(MESuSiE_cat,levels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;), labels = c(&quot;Non&quot;, &quot;EUR&quot;, &quot;AFR&quot;, &quot;Shared&quot;, &quot;Paintor&quot;)))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP,cat)
Paintor_plot_data&lt;-candidate_region%&gt;%mutate(r2 = r2_EUR,PIP = Paintor_PIP,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(Paintor_cat,levels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;), labels = c(&quot;Non&quot;, &quot;EUR&quot;, &quot;AFR&quot;, &quot;Shared&quot;, &quot;Paintor&quot;)))%&gt;%select(SNP,POS, r2,PIP,Lead_SNP,cat)

p_EUR_SuSiE&lt;-finemap_plot_fun(EUR_SuSiE_plot_data, &quot;SuSiE UKBB&quot;, &quot;PIP&quot;, 0.5)
p_AFR_SuSiE&lt;-finemap_plot_fun(AFR_SuSiE_plot_data, &quot;SuSiE GLGC&quot;, &quot;PIP&quot;, 0.5)
p_MESuSiE&lt;-finemap_plot_fun(MESuSiE_plot_data, &quot;MESuSiE&quot;, &quot;PIP&quot;, 0.5)
p_Paintor&lt;-finemap_plot_fun(Paintor_plot_data, &quot;Paintor&quot;, &quot;PIP&quot;, 0.5)

# Gene Plot
plot.range &lt;- c(min(candidate_region$POS), max(candidate_region$POS))
Gene_List_sub_coding&lt;-Gene_List%&gt;%filter(Chrom==paste0(&quot;chr&quot;,unique(candidate_region$CHR)))%&gt;%filter(Start&lt;max(candidate_region$POS),End&gt;min(candidate_region$POS))%&gt;%filter(Coding==&quot;proteincoding&quot;)%&gt;%filter(!is.na(cdsLength))%&gt;%filter(GeneLength&gt;15000|Gene==&quot;TM6SF2&quot;)
p2&lt;-gene_range_plot_fun(Gene_List_sub_coding,plot.range)

##Combine Plot together
combined_plot&lt;-(p_EUR/p_EUR_SuSiE/p_MESuSiE/p2+plot_layout(heights = c(1,1,1,1.5))|p_AFR/p_AFR_SuSiE/p_Paintor/p2+plot_layout(heights = c(1,1,1,1.5)))+plot_layout(guides = &#39;collect&#39;)&amp;theme(legend.position = &quot;bottom&quot;)
combined_plot</code></pre>
<p><img src="figure/Lipid_711.Rmd/unnamed-chunk-7-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/Lipid_711.Rmd/unnamed-chunk-7-1.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] stringr_1.5.0       ggrepel_0.9.1       susieR_0.11.84     
 [4] coin_1.4-2          survival_3.3-1      DescTools_0.99.45  
 [7] ggbreak_0.1.1       gridExtra_2.3       VennDiagram_1.7.3  
[10] futile.logger_1.4.3 ggpmisc_0.4.7       ggpp_0.4.4         
[13] patchwork_1.1.1     tidyr_1.3.0         dplyr_1.1.2        
[16] data.table_1.14.8   ggpubr_0.6.0        ggplot2_3.4.2      
[19] workflowr_1.7.0    

loaded via a namespace (and not attached):
  [1] formatR_1.14         gld_2.6.5            sandwich_3.0-2      
  [4] readxl_1.4.2         rlang_1.1.1          magrittr_2.0.3      
  [7] git2r_0.32.0         multcomp_1.4-25      matrixStats_1.0.0   
 [10] e1071_1.7-13         compiler_4.3.1       mgcv_1.8-40         
 [13] getPass_0.2-2        callr_3.7.3          vctrs_0.6.2         
 [16] quantreg_5.95        crayon_1.5.2         pkgconfig_2.0.3     
 [19] fastmap_1.1.1        backports_1.4.1      labeling_0.4.2      
 [22] utf8_1.2.3           promises_1.2.0.1     rmarkdown_2.22      
 [25] ps_1.7.2             MatrixModels_0.5-1   purrr_1.0.1         
 [28] xfun_0.39            modeltools_0.2-23    cachem_1.0.8        
 [31] aplot_0.1.10         jsonlite_1.8.3       highr_0.10          
 [34] later_1.3.1          reshape_0.8.9        irlba_2.3.5.1       
 [37] broom_1.0.5          parallel_4.3.1       R6_2.5.1            
 [40] bslib_0.5.0          stringi_1.7.12       car_3.1-2           
 [43] boot_1.3-28.1        jquerylib_0.1.4      cellranger_1.1.0    
 [46] Rcpp_1.0.11          knitr_1.39           zoo_1.8-12          
 [49] httpuv_1.6.11        Matrix_1.5-4.1       splines_4.3.1       
 [52] tidyselect_1.2.0     rstudioapi_0.14      abind_1.4-5         
 [55] yaml_2.3.7           codetools_0.2-19     processx_3.8.0      
 [58] plyr_1.8.8           lattice_0.20-45      tibble_3.2.1        
 [61] withr_2.5.1          evaluate_0.18        gridGraphics_0.5-1  
 [64] lambda.r_1.2.4       proxy_0.4-27         pillar_1.9.0        
 [67] carData_3.0-5        whisker_0.4.1        stats4_4.3.1        
 [70] ggfun_0.0.9          generics_0.1.3       rprojroot_2.0.3     
 [73] munsell_0.5.0        scales_1.2.1         rootSolve_1.8.2.3   
 [76] class_7.3-20         glue_1.6.2           lmom_2.8            
 [79] tools_4.3.1          SparseM_1.81         ggsignif_0.6.4      
 [82] Exact_3.1            fs_1.6.2             mvtnorm_1.1-3       
 [85] libcoin_1.0-9        colorspace_2.1-0     nlme_3.1-157        
 [88] cli_3.6.1            futile.options_1.0.1 fansi_1.0.5         
 [91] expm_0.999-7         mixsqp_0.3-48        gtable_0.3.1        
 [94] rstatix_0.7.2        yulab.utils_0.0.4    sass_0.4.6          
 [97] digest_0.6.30        TH.data_1.1-2        ggplotify_0.1.0     
[100] farver_2.1.1         htmltools_0.5.5      lifecycle_1.0.3     
[103] httr_1.4.6           MASS_7.3-57         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
