<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="borangao" />

<meta name="date" content="2023-10-09" />

<title>Run MESuSiE</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MESuSiE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="installation.html">Introduction</a>
</li>
<li>
  <a href="GWAS_QC.html">Data Preparation</a>
</li>
<li>
  <a href="Run_MESuSiE.html">Run MESuSiE</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis reproduce
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Simulation.html">Simulation</a>
    </li>
    <li>
      <a href="Lipid_711.html">711 Regions of Lipid</a>
    </li>
    <li>
      <a href="Lipid_810.html">810 Regions of Lipid</a>
    </li>
    <li>
      <a href="MCHC_SCZ.html">MCHC and SCZ</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/borangao/meSuSie_Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Run MESuSiE</h1>
<h4 class="author">borangao</h4>
<h4 class="date">2023-10-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-10-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>meSuSie_Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220530code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220530code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220530)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/net/fantasia/home/borang/Susie_Mult/meSuSie_Analysis/data/MESuSiE_Example.RData
</td>
<td style="text-align:left;">
data/MESuSiE_Example.RData
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomborangaomeSuSieAnalysistreeba8db567b85c23b12dd18456c89353b3fcfc63a2targetblankba8db56a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/borangao/meSuSie_Analysis/tree/ba8db567b85c23b12dd18456c89353b3fcfc63a2" target="_blank">ba8db56</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomborangaomeSuSieAnalysistreeba8db567b85c23b12dd18456c89353b3fcfc63a2targetblankba8db56a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/borangao/meSuSie_Analysis/tree/ba8db567b85c23b12dd18456c89353b3fcfc63a2" target="_blank">ba8db56</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  data/GLGC_chr_22.txt
    Untracked:  data/MESuSiE_Example.RData
    Untracked:  data/UKBB_chr_22.txt

Unstaged changes:
    Deleted:    analysis/illustration.Rmd
    Deleted:    analysis/toy_example.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Run_MESuSiE.Rmd</code>) and HTML
(<code>docs/Run_MESuSiE.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/ba8db567b85c23b12dd18456c89353b3fcfc63a2/analysis/Run_MESuSiE.Rmd" target="_blank">ba8db56</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Update my analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/borangao/meSuSie_Analysis/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/Run_MESuSiE.html" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/62ce4b3f045d8a95de6224cb4bca6e2890f28abc/analysis/Run_MESuSiE.Rmd" target="_blank">62ce4b3</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
<td>
Update my analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="mesusie-example" class="section level1">
<h1>MESuSiE Example</h1>
<div id="data-formatting-for-mesusie" class="section level2">
<h2>1. Data Formatting for MESuSiE:</h2>
<p>Use the example from Data preparation. We provide
<code>organize data</code> and <code>organize_ld</code> function to do
the sanity check and organize the summary statistics and LD into the
MESuSiE input format.</p>
<div id="organize_gwas-function" class="section level3">
<h3><code>organize_gwas</code> Function</h3>
<p>The <code>organize_gwas</code> function harmonizes two input GWAS
datasets. Specifically, it:</p>
<ol style="list-style-type: decimal">
<li>Normalizes column names for internal processing.</li>
<li>Validates the presence of essential columns like SNP, Beta, Se, Z,
and N and computes missing values as necessary.</li>
<li>Identifies strand-ambiguous and multi-allelic SNPs.</li>
<li>Ensures that SNPs match in both datasets, subsetting to the common
set if needed.</li>
<li>Checks the consistency of REF and ALT alleles across the datasets,
adjusting signs and flipping alleles where necessary.</li>
<li>Returns the organized datasets in a list format with user-specified
names.</li>
</ol>
<p>This function ensures datasets from varied sources are compatible and
ready for subsequent analysis with MESuSiE.</p>
</div>
<div id="organize_ld-function" class="section level3">
<h3><code>organize_ld</code> Function</h3>
<p>The <code>organize_ld</code> function ensures that two LD matrices
are symmetric and compatible with provided GWAS datasets:</p>
<ol style="list-style-type: decimal">
<li>Adjusts the LD matrices to be symmetric if they aren’t already.</li>
<li>Ensures there are no NA values within the LD matrices.</li>
<li>Validates that the LD matrices have SNP names in their column names
and are consistent with the GWAS data.</li>
<li>Returns the organized LD matrices in a list format with names
consistent with the GWAS datasets.</li>
</ol>
<pre class="r"><code>library(dplyr)
library(MESuSiE)
load(&quot;/net/fantasia/home/borang/Susie_Mult/meSuSie_Analysis/data/MESuSiE_Example.RData&quot;)
summ_stat_list&lt;-organize_gwas(UKBB_example%&gt;%rename(SNP = CHR_POS),GLGC_example%&gt;%rename(SNP = CHR_POS),c(&quot;EUR&quot;,&quot;AFR&quot;))
colnames(WB_cov)&lt;-UKBB_example$CHR_POS
colnames(BB_cov)&lt;-GLGC_example$CHR_POS
LD_list&lt;-organize_ld(WB_cov,BB_cov,summ_stat_list)</code></pre>
</div>
</div>
<div id="run-mesusie" class="section level2">
<h2>2. Run MESuSiE:</h2>
<pre class="r"><code>MESuSiE_res&lt;-meSuSie_core(LD_list,summ_stat_list,L=10)</code></pre>
<pre><code>*************************************************************

  Multiple Ancestry Sum of Single Effect Model (MESuSiE)          

   Visit http://www.xzlab.org/software.html For Update            

            (C) 2022 Boran Gao, Xiang Zhou                        

              GNU General Public License                          

*************************************************************
# Start data processing for sufficient statistics 
# Create MESuSiE object 
# Start data analysis 

# Data analysis is done, and now generates result 

Potential causal SNPs with PIP &gt; 0.5:  22_21958872 

Credible sets for effects: 
$cs
$cs$L1
 [1]  7  8 26 38 40 41 57 61 62 69 71 79 82


$cs_category
       L1 
&quot;EUR_AFR&quot; 

$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    0.9978499     0.9991224       0.9992341

$cs_index
[1] 1

$coverage
[1] 0.9551583

$requested_coverage
[1] 0.95


 Use MESuSiE_Plot() for visualization
# Total time used for the analysis: 0.07 mins</code></pre>
<div id="mesusie-result-interpretation" class="section level3">
<h3>MESuSiE Result Interpretation:</h3>
<p>The MESuSiE results can be interpreted in terms of the 95% credible
set and SNP-level Posterior Inclusion Probabilities (PIPs).</p>
<ul>
<li><strong>95% credible set</strong> MESuSiE’s 95% credible set
comprises SNPs with a non-zero effect in at least one ancestry. These
SNPs are further categorized as either “shared” or “ancestry-specific”
based on their contribution to the credible set.</li>
</ul>
<p>For instance, in the provided example, the credible set contains 13
SNPs labeled as <code>EUR_AFR</code>, indicating these SNPs are shared
across multiple ancestries.</p>
<ul>
<li><p><strong>PIPs</strong> We used a PIP cutoff of 0.5 to declare
signal.</p>
<ul>
<li><strong>PIP for SNP have non-zero effect in at least one
ancestry</strong></li>
</ul>
<pre class="r"><code>MESuSiE_res$pip[MESuSiE_res$cs$cs$L1]</code></pre>
<pre><code> [1] 0.02617149 0.02411160 0.04600253 0.03558618 0.02805318 0.03538103
 [7] 0.02041509 0.05667729 0.05446116 0.04994392 0.52385726 0.02241395
[13] 0.03208367</code></pre>
<p>In the example provided, one SNP exhibits a PIP greater than 0.5.
However, this does not inform us whether the SNP is shared or
ancestry-specific. To make that inference, we need to further analyze
the PIP values for shared or ancestry-specific effects, which is the
unique feature provided by MESuSiE.</p>
<ul>
<li><strong>PIP for shared or ancestry-specific effect</strong></li>
</ul>
<pre class="r"><code>MESuSiE_res$pip_config[MESuSiE_res$cs$cs$L1,]</code></pre>
<pre><code>               EUR          AFR    EUR_AFR
 [1,] 0.0003748811 0.0002535134 0.02617115
 [2,] 0.0003748810 0.0002532205 0.02411128
 [3,] 0.0003748798 0.0002476710 0.04600234
 [4,] 0.0003748806 0.0002482805 0.03558589
 [5,] 0.0003748802 0.0002483030 0.02805295
 [6,] 0.0003748805 0.0002483253 0.03538074
 [7,] 0.0003748796 0.0002484404 0.02041492
 [8,] 0.0003748794 0.0002484522 0.05667714
 [9,] 0.0003748793 0.0002485671 0.05446102
[10,] 0.0003748793 0.0002480695 0.04994379
[11,] 0.0003748793 0.0002610915 0.52385712
[12,] 0.0003748795 0.0002468203 0.02241380
[13,] 0.0003748793 0.0002458986 0.03208353</code></pre>
<p>From the results, SNP with a PIP greater than 0.5 represents a shared
causal signal.</p></li>
</ul>
</div>
</div>
<div id="configuring-tuning-parameters" class="section level2">
<h2>3.Configuring Tuning Parameters</h2>
<p>When running <code>meSuSie_core</code>, various tuning parameters can
be adjusted to refine the results:</p>
<ul>
<li><strong>Number of Effects (<code>L</code>)</strong>: Specify the
number of effects to consider.</li>
<li><strong>Prior Weights (<code>prior_weights</code>)</strong>: This
represents the prior probability of a SNP being causal. By default, it
is set to . Users can adjust this parameter to incorporate functional
annotation into MESuSiE.</li>
<li><strong>Ancestry Weight (<code>ancestry_weight</code>)</strong>:
Configure the weighting for different ancestries. We set the ratio of
ancestry-specific to shared as 3:1 to encourage the ancestry-specific
causal SNP detection.
<ul>
<li><strong>Pure shared signal:</strong> ancestry weight can be set to
c(0,0,1)</li>
<li><strong>Encourage Ancestry-specific signal</strong> ancestry weight
can be set to c(5,5,1)/sum(c(5,5,1))</li>
</ul></li>
<li><strong>Estimate Residual Variance
(<code>estimate_residual_variance</code>)</strong>: This parameter can
be set to either TRUE or FALSE. When set to TRUE, the residual variance
is estimated, while when set to FALSE, it is fixed at one. For
multi-ancestry GWAS fine-mapping, it’s typically advisable to fix the
residual variance at one to enhance robustness, given that the
heritability of the local region is often minimal. However, when using
MESuSiE for multi-ancestry eQTL fine-mapping, it’s recommended to
estimate the residual variance since the heritability of the gene
expression is relatively significant.</li>
<li><strong>Correlation Method and Threshold
(<code>cor_method &amp; cor_threshold</code>)</strong>:
<ul>
<li><strong>cor_method</strong>: Choose the method to determine the 95%
credible set based on SNP correlations within the region. Commonly used
methods include determining the minimum, mean, or median of the absolute
correlation values. By default, this is set to
<code>min_abs_corr</code>, representing the minimum absolute correlation
permissible within a credible set across ancestries. This is a prevalent
threshold for genotype data in genetic studies.</li>
<li><strong>cor_threshold</strong>: Define the cutoff for
<code>min_abs_corr</code>. By default, this is set to 0.5.</li>
</ul></li>
</ul>
<blockquote>
<p><strong>Note</strong>: We’ve made available function
<code>meSuSie_get_cs</code> allowing users to tweak the results based on
varying <code>cor_method</code> and <code>cor_threshold</code>. The best
part? You can adjust without having to rerun the entire analysis.</p>
</blockquote>
</div>
<div id="mesusie-visulization" class="section level2">
<h2>4. MESuSiE Visulization:</h2>
<pre class="r"><code>MESuSiE_Plot(MESuSiE_res,LD_list,summ_stat_list)</code></pre>
<p><img src="figure/Run_MESuSiE.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/borangao/meSuSie_Analysis/blob/504f3a9e75d5b5a4b027e6b3eb4de077680210c8/docs/figure/Run_MESuSiE.Rmd/unnamed-chunk-5-1.png" target="_blank">504f3a9</a>
</td>
<td>
borangao
</td>
<td>
2023-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] MESuSiE_1.0     dplyr_1.1.2     workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] gtable_0.3.1             xfun_0.39                bslib_0.5.0             
 [4] ggplot2_3.4.2            processx_3.8.0           ggrepel_0.9.1           
 [7] lattice_0.20-45          callr_3.7.3              vctrs_0.6.2             
[10] tools_4.3.1              ps_1.7.2                 generics_0.1.3          
[13] tibble_3.2.1             fansi_1.0.5              highr_0.10              
[16] pkgconfig_2.0.3          Matrix_1.5-4.1           data.table_1.14.8       
[19] lifecycle_1.0.3          compiler_4.3.1           farver_2.1.1            
[22] stringr_1.5.0            git2r_0.32.0             progress_1.2.2          
[25] munsell_0.5.0            getPass_0.2-2            httpuv_1.6.11           
[28] htmltools_0.5.5          sass_0.4.6               yaml_2.3.7              
[31] later_1.3.1              pillar_1.9.0             nloptr_2.0.3            
[34] crayon_1.5.2             jquerylib_0.1.4          whisker_0.4.1           
[37] tidyr_1.3.0              ellipsis_0.3.2           cachem_1.0.8            
[40] tidyselect_1.2.0         digest_0.6.30            stringi_1.7.12          
[43] purrr_1.0.1              labeling_0.4.2           RcppArmadillo_0.11.1.1.0
[46] cowplot_1.1.1            rprojroot_2.0.3          fastmap_1.1.1           
[49] grid_4.3.1               colorspace_2.1-0         cli_3.6.1               
[52] magrittr_2.0.3           utf8_1.2.3               withr_2.5.1             
[55] prettyunits_1.2.0        scales_1.2.1             promises_1.2.0.1        
[58] rmarkdown_2.22           httr_1.4.6               hms_1.1.2               
[61] evaluate_0.18            knitr_1.39               irlba_2.3.5.1           
[64] rlang_1.1.1              Rcpp_1.0.11              mixsqp_0.3-48           
[67] glue_1.6.2               rstudioapi_0.14          jsonlite_1.8.3          
[70] R6_2.5.1                 fs_1.6.2                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
