---
title: "MESuSiE"
author: "Boran Gao"
date: "2022-05-30"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---
# MESuSiE Overview
![](Fig_1.png)

MESuSiE is a multiple ancestry extension of Sum of Single Effect model for GWAS fine-mapping.MESuSiE relies on GWAS summary statistics from multiple ancestries, properly accounts for the LD structure of the focal genomic region in multiple ancestries, and explicitly models both shared and ancestry-specific causal signals to accommodate causal effect size similarity as well as heterogeneity across ancestries. MESuSiE outputs posterior inclusion probability of variant being shared or ancestry-specific causal variants. 



# MESuSiE Model 
MESuSiE builds upon the sum of single effect model, and extend the normal assumption on the effec size of causal SNP to multivariate normal, and thus capture the correlation across ancestries.
fashion:
$$
  \left(\begin{matrix}{y}_\mathbf{1}\\{y}_\mathbf{2}\\\end{matrix}\right)=\left[\begin{matrix}{X}_\mathbf{1}&0\\0&{X}_\mathbf{2}\\\end{matrix}\right]\ast\left(\begin{matrix}{b}_\mathbf{1}\\{b}_\mathbf{2}\\\end{matrix}\right)+\left(\begin{matrix}{\epsilon}_\mathbf{1}\\{\epsilon}_\mathbf{2}\\\end{matrix}\right), 
$$
In the above equation, $y_1,y_2$ are standardized phenotype for each ancestry, $X_1,X_2$ are centered genotype matrices, $b_1,b_2$ are sum of the single effect models which are in the form, 
$$
\left(\begin{matrix}{b}_\mathbf{1}\\{b}_\mathbf{2}\\\end{matrix}\right) = \sum_l \gamma_l\bigotimes \left(z_l\cdot\left(\begin{matrix}{\beta}_\mathbf{1l}\\{\beta}_\mathbf{2l}\\\end{matrix}\right) \right)
$$
which represents the summation of the single effects.


**Note**: Our software is intended strictly for research purposes. We firmly stand against any form of racial discrimination and advocate for the ethical use of MESuSiE.