---
title: "Lipid 711"
author: "borangao"
date: "2023-10-09"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## 711 Regions of 4 lipid traits

> **Note**: all the code and analysis reproduced here can be found in [Repository](https://zenodo.org/deposit/8411004#)

## 1. Feature of 95% credible set

### 1a. Set size and eQTL enrichment of 95% credible set

```{r,message=FALSE,warning=FALSE}
library(ggpubr)
library(data.table)
library(dplyr)
library(tidyr)
library(ggplot2)
library(patchwork)
library(ggpmisc)
library(VennDiagram)
library(gridExtra)
library(ggbreak)
library(DescTools)
library(coin)
library(susieR)
library(ggrepel)
library(stringr)


load("/net/fantasia/home/borang/Susie_Mult/Revision_Round_1/01_06_Real_Data/summary_res/res.RData")
custom_theme <- function() {
  theme(
    axis.text.x = element_text(size = 5),
    axis.text.y = element_text(size = 5),  
    axis.title.x = element_text(size = 7, face="bold"),
    axis.title.y = element_text(size = 7, face="bold"),
    strip.text.x = element_text(size = 5),
    strip.text.y = element_text(size = 5),
    strip.background = element_blank(),
    legend.text = element_text(size=7),
    legend.title = element_text(size=7, face="bold"),
    plot.title = element_text(size=7, hjust = 0.5),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank(), 
    axis.line = element_line(color = "black")
  )
}
################################################
#
#		Set Size/Z-score/eQTL 
#
#
###############################################
################################################
#
#		Set SiZe Part
#
###############################################		
###Median set size by Trait
all_sets_info<-data.frame(res_all%>%group_by(Trait,Region) %>% summarise(across(c("MESuSiE_cs", "SuSiE_cs","Paintor_cs"), ~ sum(.x, na.rm = TRUE))))%>%filter(MESuSiE_cs!=0, SuSiE_cs!=0, Paintor_cs!=0) ###Median Set Size across all locus
all_sets_info_long<-all_sets_info%>%pivot_longer(!(Trait|Region), names_to = "Method", values_to = "Count")
all_sets_info_long$Method<-factor(all_sets_info_long$Method,levels=c("MESuSiE_cs","SuSiE_cs","Paintor_cs"))
levels(all_sets_info_long$Method)<-c("MESuSiE","SuSiE","Paintor")

p_set = ggplot(data =all_sets_info_long,aes(x = Trait, y=Count,fill=Method))+geom_boxplot(aes(x = Trait,fill=Method),outlier.size = 0.1,fatten = 0.5,color = "darkgray")+scale_fill_manual(values=c("MESuSiE"="#023e8a","SuSiE"="#2a9d8f","Paintor"="#f4a261"),guide=FALSE)
p_set =p_set + theme_bw() + xlab("") +ylab("Set Size")+coord_cartesian(ylim=c(0,175))
p_set= p_set+custom_theme()

################################################
#
#		Z-score Part
#
###############################################		
MESuSiE_cs_Z<-res_all%>%group_by(Trait) %>%filter(MESuSiE_cs==1)%>%summarise(zmax = median(pmax(abs(zscore_WB),abs(zscore_BB))))
SuSiE_cs_Z<-res_all%>%group_by(Trait) %>%filter(SuSiE_cs==1)%>%summarise(zmax =median(pmax(abs(zscore_WB),abs(zscore_BB))))%>%pull(zmax)
Paintor_cs_Z<-res_all%>%group_by(Trait) %>%filter(Paintor_cs==1)%>%summarise(zmax = median(pmax(abs(zscore_WB),abs(zscore_BB))))%>%pull(zmax)
set_size_z_info<-data.frame(cbind(MESuSiE_cs_Z,SuSiE_cs_Z,Paintor_cs_Z))
colnames(set_size_z_info)<-c("Trait",c("MESuSiE","SuSiE","Paintor"))
set_size_z_info_long<-set_size_z_info %>%pivot_longer(!(Trait), names_to = "Method", values_to = "Z")%>%mutate(Method = factor(Method, levels=c("MESuSiE","SuSiE","Paintor")))

p_z = ggplot(data = set_size_z_info_long,aes(x = Trait, y=Z,fill=Method))+geom_bar( stat = "identity",position="dodge")+scale_fill_manual(values=c("MESuSiE"="#023e8a","SuSiE"="#2a9d8f","Paintor"="#f4a261"))
p_z = p_z + geom_text(label = round(set_size_z_info_long$Z,2),position = position_dodge(width = 1),vjust=-0.5,size = 5*5/14)
p_z = p_z + theme_bw() + xlab("") +ylab("Median |Z|")+ ylim(0,max(round(set_size_z_info_long$Z,2)+1))
p_z = p_z +custom_theme()
################################################
#
#		eQTL enrichment 
#
#
###############################################	
 ann_col_name<-c("missense", "synonymous", "utr_comb", "promotor", "CRE","liver_ind_eQTL")
  # Functions for calculating fold enrichment
  calc_fold_enrichment <- function(df, cs_col, ann_col_name) {
    df %>%
      group_by(Region) %>%
      filter(sum(!!sym(cs_col)) != 0) %>%
      group_by(Trait, !!sym(cs_col)) %>%
      summarise(across(ann_col_name, ~ sum(.x, na.rm = TRUE) / n())) %>%
      group_by(Trait) %>%
      summarise(across(ann_col_name, ~ .x[!!sym(cs_col) == 1] / .x[!!sym(cs_col) == 0]))
  }
  
  MESuSiE_PIP_ann <- calc_fold_enrichment(res_all, "MESuSiE_cs", ann_col_name)
  SuSiE_PIP_ann <- calc_fold_enrichment(res_all, "SuSiE_cs", ann_col_name)
  Paintor_PIP_ann <- calc_fold_enrichment(res_all, "Paintor_cs", ann_col_name)
  # Combine results
  Trait_CS_enrichment <- bind_rows(
    MESuSiE_PIP_ann %>% mutate(Method = "MESuSiE"),
    SuSiE_PIP_ann %>% mutate(Method = "SuSiE"),
    Paintor_PIP_ann %>% mutate(Method = "Paintor")
  ) %>% mutate(Method = factor(Method, levels = c("MESuSiE", "SuSiE", "Paintor")))%>%
    dplyr::select(Trait,liver_ind_eQTL ,Method )%>%dplyr::rename(eQTL = liver_ind_eQTL)
  # Pivot to long format
  Trait_CS_enrichment_long <- Trait_CS_enrichment %>%
    pivot_longer(cols = -c(Method, Trait), names_to = "Cat", values_to = "Prop") %>%
    mutate(Method = factor(Method, levels = c("MESuSiE", "SuSiE", "Paintor")))

p_eQTL <- ggplot(Trait_CS_enrichment_long, aes(x = Trait, y = Prop, fill = Method)) +
  geom_bar(stat = "identity", position = "dodge") +scale_fill_manual(values = c("MESuSiE" = "#023e8a", "SuSiE" = "#2a9d8f", "Paintor" = "#f4a261")) +
  geom_text(,label = round(Trait_CS_enrichment_long$Prop,2),position = position_dodge(width = 1),vjust=-0.5,size = 5*5/14)+
  xlab("") + ylab("eQTL Fold Enrichment") + ylim(0,max(round(Trait_CS_enrichment_long$Prop))+1)+
  theme_bw() + custom_theme()

p_out<-p_set/p_z/p_eQTL+plot_annotation(tag_levels = 'a')+plot_layout(guides = "collect",heights = c(1.5,1,1))&theme(legend.position = 'bottom',plot.tag = element_text(size = 7,face="bold"))
p_out
```

### 1b. Functional enrichment of 95% credible set and top signals

```{r,message=FALSE,warning=FALSE}

################################################################################
#
#
#     Functional Annotation enrichment for 95% credible set SNPS
#
#
################################################################################
 # Enrichment of 95% credible set without by trait
  calc_fold_enrichment_marginal<-function(df, cs_col, ann_col_name) {
    df %>%group_by(Region) %>%
      filter(sum(!!sym(cs_col)) != 0) %>%
      group_by( !!sym(cs_col)) %>%
      summarise(across(ann_col_name, ~ sum(.x, na.rm = TRUE) / n())) %>%
      summarise(across(ann_col_name, ~ .x[!!sym(cs_col) == 1] / .x[!!sym(cs_col) == 0]))
  }
  MESuSiE_PIP_ann <- calc_fold_enrichment_marginal(res_all, "MESuSiE_cs", ann_col_name)
  SuSiE_PIP_ann <- calc_fold_enrichment_marginal(res_all, "SuSiE_cs", ann_col_name)
  Paintor_PIP_ann <- calc_fold_enrichment_marginal(res_all, "Paintor_cs", ann_col_name)
  
  CS_enrichment <- bind_rows(
    MESuSiE_PIP_ann %>% mutate(Method = "MESuSiE"),
    SuSiE_PIP_ann %>% mutate(Method = "SuSiE"),
    Paintor_PIP_ann %>% mutate(Method = "Paintor")
  ) %>% mutate(Method = factor(Method, levels = c("MESuSiE", "SuSiE", "Paintor")))%>% 
    dplyr::rename(Missense = missense ,Synonymous = synonymous,UTR = utr_comb,Promotor = promotor,eQTL = liver_ind_eQTL)
  
  # Pivot to long format
  CS_enrichment_long <- CS_enrichment %>%
    pivot_longer(cols = -c(Method), names_to = "Cat", values_to = "Prop") %>%
    mutate(Method = factor(Method, levels = c("MESuSiE", "SuSiE", "Paintor"))) %>%
    mutate(Cat = factor(Cat, levels = c("Missense", "Synonymous", "UTR", "Promotor", "CRE","eQTL")))%>%
    mutate(Prop = round(Prop, 2))

p_set <- ggplot(data = CS_enrichment_long,aes(x = Cat, y = Prop, fill = Method)) +
  geom_col(position = "dodge") + scale_fill_manual(values = c("MESuSiE" = "#023e8a", "SuSiE" = "#2a9d8f", "Paintor" = "#f4a261")) +
  geom_text(aes(x=Cat,group=Method,y=Prop,label=Prop),position = position_dodge(width = 1),vjust=-0.5,size = 5/14*5) + 
  geom_hline(yintercept = 1, linetype = "dashed") + 
  xlab("") + ylab("Fold Enrichment Credible Set") +ylim(0,round(max(CS_enrichment_long$Prop))+1) +
  theme_bw() + custom_theme()
################################################################################
#
#
#     Functional Annotation enrichment for top 500 PIP SNPs
#
#
################################################################################	
  # Enrichment of top 500 signal
  res_all<-res_all%>%mutate(Paintor_Signal = ifelse(Paintor_PIP>0.5,1,0))
  res_all<-res_all%>%mutate(SuSiE_Signal = case_when(
    SuSiE_WB>0.5&SuSiE_BB>0.5~1,
    SuSiE_WB>0.5&SuSiE_BB<0.5~2,
    SuSiE_WB<0.5&SuSiE_BB>0.5~3,
    .default =0))
  res_all<-res_all%>%mutate(MESuSiE_Signal = case_when(
    MESuSiE_PIP_Shared>0.5~1,
    MESuSiE_PIP_WB>0.5~2,
    MESuSiE_PIP_BB>0.5~3,
    .default =0))
  top_N_signal = 500
  bg_an<-res_all%>%summarise(across(ann_col_name,~ sum(.x, na.rm = TRUE)/n()))
  MESuSiE_Signal_ann<-res_all%>%filter(MESuSiE_Signal!=0)%>% arrange(desc(MESuSiE_PIP_Either))%>%top_n(n = top_N_signal, wt = MESuSiE_PIP_Either)%>%summarise(across(ann_col_name,~ sum(.x, na.rm = TRUE)/n()))/bg_an
  SuSiE_Signal_ann<-res_all%>%filter(SuSiE_Signal!=0)%>% arrange(desc(SuSiE_PIP))%>%top_n(n = top_N_signal, wt = SuSiE_PIP)%>%summarise(across(ann_col_name,~ sum(.x, na.rm = TRUE)/n()))/bg_an
  Paintor_Signal_ann<-res_all%>%filter(Paintor_Signal!=0) %>% arrange(desc(Paintor_PIP))%>%top_n(n = top_N_signal, wt = Paintor_PIP)%>%summarise(across(ann_col_name,~ sum(.x, na.rm = TRUE)/n()))/bg_an
  
  Signal_enrichment <- bind_rows(
    MESuSiE_Signal_ann %>% mutate(Method = "MESuSiE"),
    SuSiE_Signal_ann %>% mutate(Method = "SuSiE"),
    Paintor_Signal_ann %>% mutate(Method = "Paintor")
  ) %>% mutate(Method = factor(Method, levels = c("MESuSiE", "SuSiE", "Paintor")))%>% 
    dplyr::rename(Missense = missense ,Synonymous = synonymous,UTR = utr_comb,Promotor = promotor,eQTL = liver_ind_eQTL)
  # Pivot to long format
  Signal_enrichment_long <- Signal_enrichment %>%
    pivot_longer(cols = -c(Method), names_to = "Cat", values_to = "Prop") %>%
    mutate(Method = factor(Method, levels = c("MESuSiE", "SuSiE", "Paintor"))) %>%
    mutate(Cat = factor(Cat, levels = c("Missense", "Synonymous", "UTR", "Promotor", "CRE","eQTL")))%>%
    mutate(Prop = round(Prop, 2))

p_signal <- ggplot(data = Signal_enrichment_long,aes(x = Cat, y = Prop, fill = Method)) +
  geom_col(position = "dodge") + scale_fill_manual(values = c("MESuSiE" = "#023e8a", "SuSiE" = "#2a9d8f", "Paintor" = "#f4a261")) +
  geom_text(aes(x=Cat,group=Method,y=Prop,label=Prop),position = position_dodge(width = 1),vjust=-0.5,size = 5/14*5) + 
  geom_hline(yintercept = 1, linetype = "dashed") + 
  xlab("") + ylab("Fold Enrichment Top Signal") +ylim(0,round(max(Signal_enrichment_long$Prop))+1) +
  theme_bw() + custom_theme()
p_out<-p_set/p_signal+plot_annotation(tag_levels = 'a')+plot_layout(guides = "collect",heights = c(1,1))&theme(legend.position = 'bottom',plot.tag = element_text(size = 7,face="bold"))
p_out
```

## 2. Top signals with PIP > 0.5

### 2a. Proportion of shared and ancestry-specific signal

```{r,message=FALSE,warning=FALSE}
############################################################################
#
#
#              Proportion of Signal Plot
#
#
############################################################################
Signal_number<- res_all%>%group_by(Trait)%>%
  summarise(Paintor_Either_n = sum(Paintor_Signal!=0),
            SuSiE_Shared_n = sum(SuSiE_Signal==1),
            SuSiE_WB_n = sum(SuSiE_Signal==2),
            SuSiE_BB_n = sum(SuSiE_Signal==3),
            MESuSiE_Shared_n = sum(MESuSiE_Signal==1),
            MESuSiE_WB_n = sum(MESuSiE_Signal==2),
            MESuSiE_BB_n = sum(MESuSiE_Signal==3))
Signal_number<-Signal_number%>%
  pivot_longer(cols = -c(Trait), names_to = "Cat", values_to = "Num")%>%
  separate(Cat, into = c("Method", "Signal"), sep = "_", extra = "merge") %>%
  mutate(
    Method = case_when(
      str_detect(Method, "MESuSiE") ~ "MESuSiE",
      str_detect(Method, "SuSiE") ~ "SuSiE",
      str_detect(Method, "Paintor") ~ "Paintor",
      TRUE ~ Method
    ),
    Signal = case_when(
      str_detect(Signal, "BB_n") ~ "AFR",
      str_detect(Signal, "WB_n") ~ "EUR",
      str_detect(Signal, "Shared_n") ~ "Shared",
      str_detect(Signal, "Either") ~ "Either",
      TRUE ~ Signal
    )
  )
Signal_number<-Signal_number%>%group_by(Method,Trait)%>%mutate(prop = Num/sum(Num)*100,ypos = cumsum(prop)- 0.5*prop)%>%mutate(label = paste0(Signal," ",Num))
Signal_number<-Signal_number%>%mutate(Method = factor(Method, levels = c("MESuSiE","SuSiE","Paintor")),Signal = factor(Signal, levels = c("EUR","AFR","Shared","Either")))

signal_num_plot<-ggplot(Signal_number, aes(x="", y=prop, fill=Signal)) +
  geom_bar(stat="identity", width=1, color="white") +
  coord_polar("y", start=0) +
  theme_void() + 
  theme(legend.position="none") +
  geom_text(aes(y = ypos, label = label),  size=7/14*5,color="white") +
  scale_fill_manual(values=c("#ABDB9F","#F2C1B6","#6162B0","gray"))+
  facet_grid(vars(Trait),vars(Method),labeller=label_parsed)+theme(
    strip.text.x = element_text(size = 7,face="bold"),
    strip.text.y = element_text(size = 7,face="bold"),
    strip.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank())
signal_num_plot
```

### 2b. Feature of shared and ancestry-specific signal

```{r,message=FALSE,warning=FALSE}
###########################################################################################
#
#
#         Correlation of ancestry-specific signal by MESuSiE and SuSiE
#
#
#########################################################################################
cor_ancestry_MESuSiE <- res_all %>%
    filter(MESuSiE_Signal%in%c(2,3)) %>% filter(!(SuSiE_Signal%in%c(2,3)))%>%
    mutate(Method = "MESuSiE") %>%
    dplyr::select(Beta_WB, Beta_BB, Trait, Method)
  cor_ancestry_SuSiE <- res_all %>%
    filter(SuSiE_Signal%in%c(2,3)) %>%filter(!(MESuSiE_Signal%in%c(2,3)))%>%
    mutate(Method = "SuSiE") %>%
    dplyr::select(Beta_WB, Beta_BB, Trait, Method)
  cor_ancestry_both <- res_all %>%filter(MESuSiE_Signal%in%c(2,3),SuSiE_Signal%in%c(2,3))%>%
    mutate(Method = "Both") %>%
    dplyr::select(Beta_WB, Beta_BB, Trait, Method)
  cor_ancestry <- rbind(cor_ancestry_MESuSiE, cor_ancestry_SuSiE,cor_ancestry_both)
  cor_ancestry<-cor_ancestry%>%mutate(Method = factor(Method,levels = c("MESuSiE","SuSiE","Both")))	

betaplot<- ggscatter(cor_ancestry, x = "Beta_WB", y = "Beta_BB",
                add = "reg.line",  # Add regressin line
                add.params = list(color = "red", fill = "lightgray",size = 0.5), # Customize reg. line
                conf.int = FALSE, # Add confidence interval
                size = 0.5
)+ stat_cor(method = "pearson",aes(label = ..r.label..),size = 5/14*5)+xlab("UKBB")+ylab("GLGC")+
  theme_bw() + custom_theme()+
  facet_wrap(vars(Method),ncol=3)
betaplot

#################################################################
#
#
#               MAF and Conservative Score Part
#
#
##################################################################
################################################
#
#		Phylop/GERP Score Part
#
#
################################################		

res_all<- res_all %>%
  mutate(
    MESuSiE_Conservative = case_when(
      MESuSiE_PIP_WB>0.5|MESuSiE_PIP_BB>0.5 ~ "Ancestry-specific",
      MESuSiE_PIP_Shared>0.5 ~ "Shared",
      TRUE ~ "None"
    ),
    SuSiE_Conservative = case_when(
      (SuSiE_WB>0.5&SuSiE_BB<0.5)|(SuSiE_BB>0.5&SuSiE_WB<0.5) ~ "Ancestry-specific",
      SuSiE_Shared>0.5 ~ "Shared",
      TRUE ~ "None"
    )
  )%>%
  mutate(
    MESuSiE_Conservative = factor(MESuSiE_Conservative, levels = c("None", "Ancestry-specific", "Shared")),
    SuSiE_Conservative = factor(SuSiE_Conservative, levels = c("None", "Ancestry-specific", "Shared"))
  )



####Phylop
grouped_data_phylop <- res_all %>%
  mutate(abs_phylop = abs(phylop)) %>%
  select(MESuSiE_Conservative, abs_phylop)%>%
  group_by(MESuSiE_Conservative) %>%
  summarise(phylop_values = list(abs_phylop))%>%
  pull(phylop_values)


grouped_data_GERP <- res_all %>%
  mutate(GERP = abs(GERP)) %>%
  select(MESuSiE_Conservative, GERP)%>%
  group_by(MESuSiE_Conservative) %>%
  summarise(GERP_values = list(GERP))%>%
  pull(GERP_values)


#########Conservation Plot
conserve_data<-res_all%>%mutate(Signal =MESuSiE_Conservative, Phylop =abs(phylop) )%>%select(Signal,Phylop,GERP)
conserve_data_long<-conserve_data%>%pivot_longer(!(Signal), names_to = "Method", values_to = "Score")
conserve_data_long$Method<-factor(conserve_data_long$Method,levels = c("Phylop","GERP"))
df <-conserve_data_long %>%group_by(Method,Signal) %>%summarize(ymin = min(Score,na.rm = T), lower = quantile(Score, .25,na.rm = T),  middle = mean(Score,na.rm = T), upper = quantile(Score, .75,na.rm = T),ymax = max(Score,na.rm = T)) 
p_set_phylop = df %>% ggplot(aes(x = factor(Method), fill=Signal)) + geom_boxplot(fatten = 2,aes(ymin = ymin, ymax = ymax, lower = lower, upper = upper, middle = middle), stat = 'identity')+scale_fill_manual(values=c("None"="#DAFFED","Ancestry-specific"="#fffbdb","Shared" = "#7776bc"))
p_set_phylop =p_set_phylop + theme_bw() + xlab("") +ylab("Conservation Score")
p_set_phylop= p_set_phylop  + custom_theme()
p_set_phylop= p_set_phylop+ theme(legend.position="bottom")		

################################################
#
#		MAF Part
#
#
################################################	
 maf_shared<-data.frame(res_all)%>%filter(MESuSiE_Signal==1)%>%mutate(MAF_diff = ifelse(MA_WB==MA_BB,MAF_WB-MAF_BB,1-MAF_WB-MAF_BB))%>%pull(MAF_diff)
  maf_ancestry<-data.frame(res_all)%>%filter(MESuSiE_Signal%in%c(2,3))%>%mutate(MAF_diff = ifelse(MA_WB==MA_BB,MAF_WB-MAF_BB,1-MAF_WB-MAF_BB))%>%pull(MAF_diff)
 

maf_data_plot<-data.frame("MAF" = c(maf_shared,maf_ancestry),"Signal"=c(rep("Shared",length(maf_shared)),rep("Ancestry-specific",length(maf_ancestry))))
maf_data_plot$Signal<-factor(maf_data_plot$Signal)
maf_data_plot$Signal<-factor(maf_data_plot$Signal,levels = c(levels(maf_data_plot$Signal),"None"))

p_set = ggplot(data =maf_data_plot,aes(x = factor(Signal), y=MAF,fill=Signal))+geom_boxplot(aes(x = factor(Signal),fill=Signal),outlier.size = 0.01) +scale_fill_manual(values=c("None"="#DAFFED","Ancestry-specific"="#fffbdb","Shared" = "#7776bc"),guide=FALSE)
p_set = p_set + stat_compare_means(size= 5/14*5)
p_set =p_set + theme_bw() + xlab("") +ylab("MAF difference")
p_set= p_set + custom_theme()
p_set= p_set + theme(legend.position="none")	
p_out<-p_set+p_set_phylop+plot_annotation(tag_levels = 'a') + plot_layout(guides = "collect") & theme(legend.position = "bottom",plot.tag = element_text(size = 7,face="bold"))
p_out
```

## 3. Example locus

### 3a. FAD2 Example

```{r,message=FALSE,warning=FALSE,fig.height= 9,fig.width=8}
##############################################################
#
#
#       Real Data Example Plotter
#
#
################################################################

gwas_plot_fun <- function(data_plot, xlab_name, ylab_name, yintercept) {
  
  p_manhattan = ggplot() + geom_point(data = data_plot%>%filter(Lead_SNP==0), aes(x = POS, y = PIP, color = r2), size = 1)
  p_manhattan = p_manhattan + geom_point(data = data_plot%>%filter(Lead_SNP==1), aes(x = POS, y = PIP), size = 1.5, color = "red") +
    geom_text_repel(data = data_plot%>%filter(Lead_SNP==1), mapping = aes(x = POS, y = PIP, label = SNP), vjust = 1.2, size = 7/14*5, show.legend =FALSE) 
  p_manhattan = p_manhattan +
    scale_color_stepsn(
      colors = c("navy", "lightskyblue", "green", "orange", "red"),
      breaks = seq(0.2, 0.8, by = 0.2),
      limits = c(0, 1),
      show.limits = TRUE,
      na.value = 'grey50',
      name = expression(R^2)
    )
  p_manhattan = p_manhattan +
    geom_hline(
      yintercept = yintercept,
      linetype = "dashed",
      color = "grey50",
      size = 0.5
    ) 
  p_manhattan = p_manhattan +
    geom_vline(
      xintercept = data_plot%>%filter(lead_SNP==1)%>%pull(POS),
      linetype = "dashed",
      color = "grey50",
      size = 0.5
    ) 
  p_manhattan = p_manhattan + xlim(min(data_plot$POS),max(data_plot$POS))
  p_manhattan = p_manhattan + expand_limits(x = round(max(data_plot$POS)/1e6)*1e6)
  if(max(data_plot$POS>1e6)){
    p_manhattan = p_manhattan + scale_x_continuous(labels = function(x) paste0(x / 1e6, " MB"))
  }
  if(max(data_plot$POS<1e6)){
    p_manhattan = p_manhattan + scale_x_continuous(labels = function(x) paste0(x / 1e3, " KB"))
  }
  p_manhattan = p_manhattan + xlab(xlab_name) +ylab(ylab_name)
  p_manhattan = p_manhattan + guides(fill = guide_legend(title = as.expression(bquote(R^2))))
  p_manhattan = p_manhattan + theme_bw()+custom_theme()
  return(p_manhattan)
}

###Function used for PIP plot	
finemap_plot_fun<-function(data_plot,xlab_name,ylab_name,yintercept){
  p_manhattan = ggplot() + geom_point(data = data_plot, aes(x = POS, y = PIP, color = r2,shape = cat))+scale_shape_manual(name="Category",drop=FALSE,values=c(20,24,25,23,22))
  p_manhattan = p_manhattan + geom_text_repel(data =data_plot%>%filter(Lead_SNP==1), mapping=aes(x=POS, y=PIP, label=SNP),vjust=1.2, size= 7/14*5,show.legend = FALSE)
  p_manhattan = p_manhattan + theme_bw()+scale_color_stepsn(
    colors = c("navy", "lightskyblue", "green", "orange", "red"),
    breaks = seq(0.2, 0.8, by = 0.2),
    limits = c(0, 1),
    show.limits = TRUE,
    na.value = 'grey50',
    name = expression(R^2)
  )
  p_manhattan = p_manhattan + geom_hline(
    yintercept =yintercept,
    linetype = "dashed",
    color = "grey50",
    size = 0.5
  ) + geom_vline(
    xintercept = data_plot%>%filter(lead_SNP==1)%>%pull(POS),
    linetype = "dashed",
    color = "grey50",
    size = 0.5
  ) 
  p_manhattan = p_manhattan + xlim(min(data_plot$POS),max(data_plot$POS))
  p_manhattan = p_manhattan + expand_limits(x = round(max(data_plot$POS)/1e6)*1e6)
  if(max(data_plot$POS>1e6)){
    p_manhattan = p_manhattan + scale_x_continuous(labels = function(x) paste0(x / 1e6, " MB"))
  }
  if(max(data_plot$POS<1e6)){
    p_manhattan = p_manhattan + scale_x_continuous(labels = function(x) paste0(x / 1e3, " KB"))
  }
  p_manhattan= p_manhattan+xlab(xlab_name)+ylab(ylab_name)
  p_manhattan= p_manhattan+guides(fill=guide_legend(title=as.expression(bquote(R^2))))
  p_manhattan = p_manhattan + theme_bw()+custom_theme()
  return(p_manhattan)
}			
# Function used for gene plot 
gene_range_plot_fun<-function(gene_list_data,plot.range){
  p<-ggplot(data = gene_list_data) +
    geom_linerange(aes(x = Gene, ymin = Start, ymax = End))+ylim(plot.range)+ expand_limits(y = round(max(plot.range[2])/1e6)*1e6)+scale_y_continuous(labels = function(y) paste0(y / 1e6, " MB"))+coord_flip()+
    geom_text(aes(x = Gene, y = Start, label = Gene), hjust = "right", size = 5/14*5) + ylab(paste0("chr",unique(gsub("chr","",gene_list_data$Chrom))))+ xlab("Gene") + 
    theme_bw() +  theme(
      axis.text.x = element_text(size = 5),
      axis.text.y = element_blank(),  
      axis.ticks.y =  element_blank(), 
      axis.title.x = element_text(size = 7, face="bold"),
      axis.title.y = element_text(size = 7, face="bold"),
      strip.text.x = element_text(size = 5),
      strip.text.y = element_text(size = 5),
      strip.background = element_blank(),
      legend.text = element_text(size=7),
      legend.title = element_text(size=7, face="bold"),
      plot.title = element_text(size=7, hjust = 0.5),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank(), 
      axis.line.x  = element_line(color = "black"),
      axis.line.y  = element_line(color = "black")
    )
  return(p)
}

#####################################################################################################################
#
#
#
#                 Example showcase
#
#
#
#####################################################################################################################
Gene_List<-fread("/net/fantasia/home/borang/Susie_Mult/simulation/simu_0120/data/Gencode_GRCh37_Genes_UniqueList2021.txt",header=T)

###################################################################
#
#
#             FADS2 independent eQTL shared signal
#
#
##################################################################
region = 384
trait_name = "TG"
ref_panel = "UKB1"
p_threshold_index = 1
res_z_dir<-paste0("/net/fantasia/home/borang/Susie_Mult/Revision_Round_1/01_02_Real_Data/",trait_name,"/",trait_name,"_REF_",ref_panel,"_P",p_threshold_index,"/")
out_dir<-paste0(res_z_dir,"data/")
out_res_dir<-paste0(res_z_dir,"res/")

##Data Reprocess
WB_COV<-as.matrix(fread(paste0(out_dir,"Region_",region,".LD1")))
BB_COV<-as.matrix(fread(paste0(out_dir,"Region_",region,".LD2")))
load(paste0(out_res_dir,"MESuSiE_region_",region,".RData"))

candidate_region<-res_all%>%filter(Region==region)
# rs174564 is the eQTL of FADS2 gene, highlighted in the paper
lead_SNP = "rs174564"
lead_SNP_index<-which(candidate_region$SNP==lead_SNP)
candidate_region<-candidate_region%>%mutate(r2_EUR  = unname(unlist((WB_COV[,lead_SNP_index])^2)) ,r2_AFR = unname(unlist((BB_COV[,lead_SNP_index])^2)),POS = as.numeric(POS))
####Category Setting
candidate_region<-candidate_region%>%mutate(SuSiE_cat = case_when(SuSiE_WB>0.5&SuSiE_BB>0.5 ~ 3,
                                                                  SuSiE_WB>0.5&SuSiE_BB<0.5 ~ 1,
                                                                  SuSiE_WB<0.5&SuSiE_BB>0.5 ~ 2,
                                                                  TRUE ~ 0),
                                            Paintor_cat = case_when(Paintor_PIP>0.5~4,
                                                                    TRUE~0),
                                            MESuSiE_cat = case_when(MESuSiE_PIP_WB>0.5~1,
                                                                    MESuSiE_PIP_BB>0.5~2,
                                                                    MESuSiE_PIP_Shared>0.5~3,
                                                                    TRUE~0))


###GWAS PLOT
EUR_GWAS_plot_data<-candidate_region%>%mutate(r2 = r2_EUR,PIP = -log10(2*pnorm(-abs(zscore_WB))),Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS))%>%select(SNP,POS, r2,PIP,Lead_SNP)
AFR_GWAS_plot_data<-candidate_region%>%mutate(r2 = r2_AFR,PIP = -log10(2*pnorm(-abs(zscore_BB))),Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS))%>%select(SNP,POS, r2,PIP,Lead_SNP)
p_EUR<-gwas_plot_fun (EUR_GWAS_plot_data, "UKBB", "-log10(P-value)", -log10(5e-8))
p_AFR<-gwas_plot_fun (AFR_GWAS_plot_data, "GLGC", "-log10(P-value)", -log10(5e-8))

###Finemap Plot
EUR_SuSiE_plot_data<-candidate_region%>%mutate(r2 = r2_EUR,PIP = SuSiE_WB,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(SuSiE_cat,levels = c("0", "1", "2", "3", "4"), labels = c("Non", "EUR", "AFR", "Shared", "Paintor")))%>%select(SNP,POS, r2,PIP,Lead_SNP,cat)
AFR_SuSiE_plot_data<-candidate_region%>%mutate(r2 = r2_AFR,PIP = SuSiE_BB,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(SuSiE_cat,levels = c("0", "1", "2", "3", "4"), labels = c("Non", "EUR", "AFR", "Shared", "Paintor")))%>%select(SNP,POS, r2,PIP,Lead_SNP,cat)
MESuSiE_plot_data<-candidate_region%>%mutate(r2 = r2_EUR,PIP = MESuSiE_PIP_Either,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(MESuSiE_cat,levels = c("0", "1", "2", "3", "4"), labels = c("Non", "EUR", "AFR", "Shared", "Paintor")))%>%select(SNP,POS, r2,PIP,Lead_SNP,cat)
Paintor_plot_data<-candidate_region%>%mutate(r2 = r2_EUR,PIP = Paintor_PIP,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(Paintor_cat,levels = c("0", "1", "2", "3", "4"), labels = c("Non", "EUR", "AFR", "Shared", "Paintor")))%>%select(SNP,POS, r2,PIP,Lead_SNP,cat)

p_EUR_SuSiE<-finemap_plot_fun(EUR_SuSiE_plot_data, "SuSiE UKBB", "PIP", 0.5)
p_AFR_SuSiE<-finemap_plot_fun(AFR_SuSiE_plot_data, "SuSiE GLGC", "PIP", 0.5)
p_MESuSiE<-finemap_plot_fun(MESuSiE_plot_data, "MESuSiE", "PIP", 0.5)
p_Paintor<-finemap_plot_fun(Paintor_plot_data, "Paintor", "PIP", 0.5)

# Gene Plot
plot.range <- c(min(candidate_region$POS), max(candidate_region$POS))
Gene_List_sub_coding<-Gene_List%>%filter(Chrom==paste0("chr",unique(candidate_region$CHR)))%>%filter(Start<max(candidate_region$POS),End>min(candidate_region$POS))%>%filter(Coding=="proteincoding")%>%filter(!is.na(cdsLength))%>%filter(GeneLength>15000)
p2<-gene_range_plot_fun(Gene_List_sub_coding,plot.range)

##Combine Plot together
combined_plot<-(p_EUR/p_EUR_SuSiE/p_MESuSiE/p2+plot_layout(heights = c(1,1,1,1.5))|p_AFR/p_AFR_SuSiE/p_Paintor/p2+plot_layout(heights = c(1,1,1,1.5)))+plot_layout(guides = 'collect')&theme(legend.position = "bottom")
combined_plot
```

### 3b. APOH Example

```{r,message=FALSE,warning=FALSE,fig.height= 9,fig.width=8}
##############################################################
#
#
#			APOH missense shared for TG (rs1801689)
#
################################################################	
region = 438
trait_name = "TG"
ref_panel = "UKB1"
p_threshold_index = 1
res_z_dir<-paste0("/net/fantasia/home/borang/Susie_Mult/Revision_Round_1/01_02_Real_Data/",trait_name,"/",trait_name,"_REF_",ref_panel,"_P",p_threshold_index,"/")
out_dir<-paste0(res_z_dir,"data/")
out_res_dir<-paste0(res_z_dir,"res/")

##Date Reprocess
WB_COV<-as.matrix(fread(paste0(out_dir,"Region_",region,".LD1")))
BB_COV<-as.matrix(fread(paste0(out_dir,"Region_",region,".LD2")))
load(paste0(out_res_dir,"MESuSiE_region_",region,".RData"))

candidate_region<-res_all%>%filter(Region==region)
# rs1801689 is the missense of APOH gene, highlighted in the paper
lead_SNP = "rs1801689"
lead_SNP_index<-which(candidate_region$SNP==lead_SNP)
candidate_region<-candidate_region%>%mutate(r2_EUR  = unname(unlist((WB_COV[,lead_SNP_index])^2)) ,r2_AFR = unname(unlist((BB_COV[,lead_SNP_index])^2)),POS = as.numeric(POS))
####Category Setting
candidate_region<-candidate_region%>%mutate(SuSiE_cat = case_when(SuSiE_WB>0.5&SuSiE_BB>0.5 ~ 3,
                                                                  SuSiE_WB>0.5&SuSiE_BB<0.5 ~ 1,
                                                                  SuSiE_WB<0.5&SuSiE_BB>0.5 ~ 2,
                                                                  TRUE ~ 0),
                                            Paintor_cat = case_when(Paintor_PIP>0.5~4,
                                                                    TRUE~0),
                                            MESuSiE_cat = case_when(MESuSiE_PIP_WB>0.5~1,
                                                                    MESuSiE_PIP_BB>0.5~2,
                                                                    MESuSiE_PIP_Shared>0.5~3,
                                                                    TRUE~0))

###GWAS PLOT
EUR_GWAS_plot_data<-candidate_region%>%mutate(r2 = r2_EUR,PIP = -log10(2*pnorm(-abs(zscore_WB))),Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS))%>%select(SNP,POS, r2,PIP,Lead_SNP)
AFR_GWAS_plot_data<-candidate_region%>%mutate(r2 = r2_AFR,PIP = -log10(2*pnorm(-abs(zscore_BB))),Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS))%>%select(SNP,POS, r2,PIP,Lead_SNP)
p_EUR<-gwas_plot_fun (EUR_GWAS_plot_data, "UKBB", "-log10(P-value)", -log10(5e-8))
p_AFR<-gwas_plot_fun (AFR_GWAS_plot_data, "GLGC", "-log10(P-value)", -log10(5e-8))

###Finemap Plot
EUR_SuSiE_plot_data<-candidate_region%>%mutate(r2 = r2_EUR,PIP = SuSiE_WB,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(SuSiE_cat,levels = c("0", "1", "2", "3", "4"), labels = c("Non", "EUR", "AFR", "Shared", "Paintor")))%>%select(SNP,POS, r2,PIP,Lead_SNP,cat)
AFR_SuSiE_plot_data<-candidate_region%>%mutate(r2 = r2_AFR,PIP = SuSiE_BB,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(SuSiE_cat,levels = c("0", "1", "2", "3", "4"), labels = c("Non", "EUR", "AFR", "Shared", "Paintor")))%>%select(SNP,POS, r2,PIP,Lead_SNP,cat)
MESuSiE_plot_data<-candidate_region%>%mutate(r2 = r2_EUR,PIP = MESuSiE_PIP_Either,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(MESuSiE_cat,levels = c("0", "1", "2", "3", "4"), labels = c("Non", "EUR", "AFR", "Shared", "Paintor")))%>%select(SNP,POS, r2,PIP,Lead_SNP,cat)
Paintor_plot_data<-candidate_region%>%mutate(r2 = r2_EUR,PIP = Paintor_PIP,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(Paintor_cat,levels = c("0", "1", "2", "3", "4"), labels = c("Non", "EUR", "AFR", "Shared", "Paintor")))%>%select(SNP,POS, r2,PIP,Lead_SNP,cat)

p_EUR_SuSiE<-finemap_plot_fun(EUR_SuSiE_plot_data, "SuSiE UKBB", "PIP", 0.5)
p_AFR_SuSiE<-finemap_plot_fun(AFR_SuSiE_plot_data, "SuSiE GLGC", "PIP", 0.5)
p_MESuSiE<-finemap_plot_fun(MESuSiE_plot_data, "MESuSiE", "PIP", 0.5)
p_Paintor<-finemap_plot_fun(Paintor_plot_data, "Paintor", "PIP", 0.5)

# Gene Plot
plot.range <- c(min(candidate_region$POS), max(candidate_region$POS))
Gene_List_sub_coding<-Gene_List%>%filter(Chrom==paste0("chr",unique(candidate_region$CHR)))%>%filter(Start<max(candidate_region$POS),End>min(candidate_region$POS))%>%filter(Coding=="proteincoding")%>%filter(!is.na(cdsLength))%>%filter(GeneLength>15000)
p2<-gene_range_plot_fun(Gene_List_sub_coding,plot.range)

##Combine Plot together
combined_plot<-(p_EUR/p_EUR_SuSiE/p_MESuSiE/p2+plot_layout(heights = c(1,1,1,1.5))|p_AFR/p_AFR_SuSiE/p_Paintor/p2+plot_layout(heights = c(1,1,1,1.5)))+plot_layout(guides = 'collect')&theme(legend.position = "bottom")
combined_plot
```


### 3c. TM6SF2 Example

```{r,message=FALSE,warning=FALSE,fig.height= 9,fig.width=8}
#######################################################################
#
#			TM6SF2 gene and LDL association with ancestry-specific effect
#
#
##########################################################################

region = 275
trait_name = "LDL"
ref_panel = "UKB1"
p_threshold_index = 1
res_z_dir<-paste0("/net/fantasia/home/borang/Susie_Mult/Revision_Round_1/01_02_Real_Data/",trait_name,"/",trait_name,"_REF_",ref_panel,"_P",p_threshold_index,"/")
out_dir<-paste0(res_z_dir,"data/")
out_res_dir<-paste0(res_z_dir,"res/")

##Date Reprocess
WB_COV<-as.matrix(fread(paste0(out_dir,"Region_",region,".LD1")))
BB_COV<-as.matrix(fread(paste0(out_dir,"Region_",region,".LD2")))
load(paste0(out_res_dir,"MESuSiE_region_",region,".RData"))

candidate_region<-res_all%>%filter(Region==region)
# rs58542926 is the missense of TM6SF2 gene gene, highlighted in the paper
lead_SNP = "rs58542926"
lead_SNP_index<-which(candidate_region$SNP==lead_SNP)
candidate_region<-candidate_region%>%mutate(r2_EUR  = unname(unlist((WB_COV[,lead_SNP_index])^2)) ,r2_AFR = unname(unlist((BB_COV[,lead_SNP_index])^2)),POS = as.numeric(POS))
####Category Setting
candidate_region<-candidate_region%>%mutate(SuSiE_cat = case_when(SuSiE_WB>0.5&SuSiE_BB>0.5 ~ 3,
                                                                  SuSiE_WB>0.5&SuSiE_BB<0.5 ~ 1,
                                                                  SuSiE_WB<0.5&SuSiE_BB>0.5 ~ 2,
                                                                  TRUE ~ 0),
                                            Paintor_cat = case_when(Paintor_PIP>0.5~4,
                                                                    TRUE~0),
                                            MESuSiE_cat = case_when(MESuSiE_PIP_WB>0.5~1,
                                                                    MESuSiE_PIP_BB>0.5~2,
                                                                    MESuSiE_PIP_Shared>0.5~3,
                                                                    TRUE~0))
###GWAS PLOT
EUR_GWAS_plot_data<-candidate_region%>%mutate(r2 = r2_EUR,PIP = -log10(2*pnorm(-abs(zscore_WB))),Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS))%>%select(SNP,POS, r2,PIP,Lead_SNP)
AFR_GWAS_plot_data<-candidate_region%>%mutate(r2 = r2_AFR,PIP = -log10(2*pnorm(-abs(zscore_BB))),Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS))%>%select(SNP,POS, r2,PIP,Lead_SNP)
p_EUR<-gwas_plot_fun (EUR_GWAS_plot_data, "UKBB", "-log10(P-value)", -log10(5e-8))
p_AFR<-gwas_plot_fun (AFR_GWAS_plot_data, "GLGC", "-log10(P-value)", -log10(5e-8))

###Finemap Plot
EUR_SuSiE_plot_data<-candidate_region%>%mutate(r2 = r2_EUR,PIP = SuSiE_WB,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(SuSiE_cat,levels = c("0", "1", "2", "3", "4"), labels = c("Non", "EUR", "AFR", "Shared", "Paintor")))%>%select(SNP,POS, r2,PIP,Lead_SNP,cat)
AFR_SuSiE_plot_data<-candidate_region%>%mutate(r2 = r2_AFR,PIP = SuSiE_BB,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(SuSiE_cat,levels = c("0", "1", "2", "3", "4"), labels = c("Non", "EUR", "AFR", "Shared", "Paintor")))%>%select(SNP,POS, r2,PIP,Lead_SNP,cat)
MESuSiE_plot_data<-candidate_region%>%mutate(r2 = r2_EUR,PIP = MESuSiE_PIP_Either,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(MESuSiE_cat,levels = c("0", "1", "2", "3", "4"), labels = c("Non", "EUR", "AFR", "Shared", "Paintor")))%>%select(SNP,POS, r2,PIP,Lead_SNP,cat)
Paintor_plot_data<-candidate_region%>%mutate(r2 = r2_EUR,PIP = Paintor_PIP,Lead_SNP = ifelse(SNP==lead_SNP,1,0),POS= as.numeric(POS),cat = factor(Paintor_cat,levels = c("0", "1", "2", "3", "4"), labels = c("Non", "EUR", "AFR", "Shared", "Paintor")))%>%select(SNP,POS, r2,PIP,Lead_SNP,cat)

p_EUR_SuSiE<-finemap_plot_fun(EUR_SuSiE_plot_data, "SuSiE UKBB", "PIP", 0.5)
p_AFR_SuSiE<-finemap_plot_fun(AFR_SuSiE_plot_data, "SuSiE GLGC", "PIP", 0.5)
p_MESuSiE<-finemap_plot_fun(MESuSiE_plot_data, "MESuSiE", "PIP", 0.5)
p_Paintor<-finemap_plot_fun(Paintor_plot_data, "Paintor", "PIP", 0.5)

# Gene Plot
plot.range <- c(min(candidate_region$POS), max(candidate_region$POS))
Gene_List_sub_coding<-Gene_List%>%filter(Chrom==paste0("chr",unique(candidate_region$CHR)))%>%filter(Start<max(candidate_region$POS),End>min(candidate_region$POS))%>%filter(Coding=="proteincoding")%>%filter(!is.na(cdsLength))%>%filter(GeneLength>15000|Gene=="TM6SF2")
p2<-gene_range_plot_fun(Gene_List_sub_coding,plot.range)

##Combine Plot together
combined_plot<-(p_EUR/p_EUR_SuSiE/p_MESuSiE/p2+plot_layout(heights = c(1,1,1,1.5))|p_AFR/p_AFR_SuSiE/p_Paintor/p2+plot_layout(heights = c(1,1,1,1.5)))+plot_layout(guides = 'collect')&theme(legend.position = "bottom")
combined_plot

```

